{% extends "base.html" %}
{% block head %}
{{ super() }}
<script type="text/javascript">
    active_menu = 'help';
    active_submenu = '';
</script>
<script type="text/javascript" src="{{ url_for('static', filename='js/help.js') }}"></script>
{% endblock %}

{% block content %}
<div class="row">
    <nav class="bg-secondary sidebar">
        <div class="position-sticky pt-3">
            <ul class="nav flex-column">
                <li class="nav-item">
                    <a class="nav-link active" aria-current="page" href="#manual">User's Manual</a>
                    <nav class="nav flex-column ms-1">
                        <a class="nav-link jump-link" aria-current="page" href="#headingDS">Datasets and Their
                            Annotations</a>
                        <a class="nav-link jump-link" aria-current="page" href="#headingSA">Standardized
                            Annotations</a>
                        <a class="nav-link jump-link" aria-current="page" href="#headingLA">List of
                            Abbreviations</a>
                        <a class="nav-link jump-link" aria-current="page" href="#headingGraphs">Graphs and
                            Search Options</a>
                    </nav>
                </li>
                <li class="nav-item mt-2">
                    <a class="nav-link" href="#faq">FAQ</a>
                    <nav class="nav subnav flex-column ms-2">
                        <a class="nav-link jump-link" aria-current="page" href="#headingAnnotations">Annotations</a>
                        <a class="nav-link jump-link" aria-current="page" href="#headingDA">Data Analysis</a>
                        <a class="nav-link jump-link" aria-current="page" href="#headingDBC">Database Contents</a>
                    </nav>
                </li>
            </ul>
        </div>
    </nav>

    <main class="main-panel">
        <h1>Help</h1>
        <div id="manual" class="jump-div">
            <h2>User's Manual</h2>
            <hr>
            <div class="p-3">
                <p>The <em>TP53</em> database contains exclusively <em>TP53</em> mutations associated with human cancers
                    that have
                    been identified by sequencing and published in the peer-reviewed literature or curated from other
                    genomic databases. </p>

                <p>Data are organized in datasets that can be <a href="search_gene_by_var">searched and analyzed</a>
                    with graph tools (see below) or fully <a href="get_tp53data">downloaded</a>. </p>

                <p>Downloaded files, annotations and graph tools are described below.</p>

                <div class="accordion" id="accordionUM">
                    <div class="accordion-item">
                        <h2 class="accordion-header" id="headingDS">
                            <button class="accordion-button" type="button" data-bs-toggle="collapse"
                                    data-bs-target="#collapseDS" aria-expanded="true" aria-controls="collapseDS">
                                Datasets and Their Annotations
                            </button>
                        </h2>
                        <div id="collapseDS" class="accordion-collapse collapse show" aria-labelledby="headingDS"
                             data-bs-parent="#accordionUM">
                            <div class="accordion-body">
                                <div>
                                    <ul class="list-group list-group-flush">
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapseGV" aria-expanded="false"
                                                        aria-controls="collapseGV">
                                                </button>
                                                Functional / Structural Data
                                            </div>
                                            <div class="collapse" id="collapseGV">
                                                <div class="bg-light p-3 my-3">
                                                    <h4>geneVariationIARC <em>TP53</em> Database,R20.txt</h4>
                                                    <div class="pb-5">
                                                        <table class="table">
                                                            <thead>
                                                            <tr>
                                                                <th>Column head</th>
                                                                <th>Description</th>
                                                            </tr>
                                                            </thead>
                                                            <tbody>
                                                            <tr>
                                                                <td><a id="MUT_id"></a>MUT_ID</td>
                                                                <td>Unique identifier of each gene variation reported in
                                                                    the database. This identifier is used in all
                                                                    datasets
                                                                    (somatic, polymorhisms, germline).
                                                                </td>
                                                            </tr>

                                                            <tr>
                                                                <td><a id="hg19_Chr17_coordinates"></a>hg19_Chr17_coordinates
                                                                </td>
                                                                <td>Chromosome coordinate of mutation: start position
                                                                    based
                                                                    on
                                                                    hg19
                                                                    human
                                                                    genome
                                                                    assembly.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="hg38_Chr17_coordinates"></a>hg38_Chr17_coordinates
                                                                </td>
                                                                <td>Chromosome coordinate of mutation: start position
                                                                    based
                                                                    on
                                                                    hg38
                                                                    human
                                                                    genome
                                                                    assembly.
                                                                </td>
                                                            </tr>

                                                            <tr>
                                                                <td><a id="ExonIntron"></a>ExonIntron</td>
                                                                <td>Location of the mutation in the introns or exons in
                                                                    <em>TP53</em>
                                                                    gene for
                                                                    the
                                                                    reference sequence NM_000546.5.
                                                                    Terms occurring in this column are "1-intron"' to
                                                                    "11-intron" and
                                                                    "2-exon"
                                                                    to "11-exon". An "i" or "e" in front mean that the
                                                                    mutation
                                                                    is
                                                                    located
                                                                    within the indicated intron
                                                                    or exon with no information on the precise location.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="Codon_number"></a>Codon_number</td>
                                                                <td>For mutations in exons, codon number
                                                                    at which the mutation is located (1-393). If a
                                                                    mutation
                                                                    spans more
                                                                    than one codon, (e.g. tandem mutation or deletion of
                                                                    several
                                                                    bases) only the first (5') codon is entered. For
                                                                    mutations
                                                                    in
                                                                    introns, 0 is entered.
                                                                </td>
                                                            </tr>
                                                            <tr>

                                                                <td><a id="WT_nucleotide"></a>WT_nucleotide</td>
                                                                <td>Base in the reference sequence at the position of
                                                                    the
                                                                    mutation on
                                                                    the
                                                                    coding
                                                                    sequence.
                                                                </td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="Mutant_nucleotide"></a>Mutant_nucleotide</td>
                                                                <td>Mutant base, described on the coding strand.</td>

                                                            </tr>
                                                            <tr>
                                                                <td><a id="Description"></a>Description</td>
                                                                <td>Nucleotide change read from the coding sequence. For
                                                                    deletions and insertions, the number of bases
                                                                    deleted
                                                                    (del)
                                                                    or
                                                                    inserted (ins) is given. For more complex mutation
                                                                    events, a
                                                                    full
                                                                    description is given as indicated in the original
                                                                    publication.
                                                                </td>
                                                            </tr>
                                                            <tr>

                                                                <td><a id="c_description"></a>c_description</td>
                                                                <td>Mutation nomenclature according to <a
                                                                        href="https://www.genomic.unimelb.edu.au/mdi/mutnomen/"
                                                                        target="_blank">HGVS</a>
                                                                    standards and using the NM_000546.5 coding sequence
                                                                    as
                                                                    reference.
                                                                </td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="g_description"></a>g_description</td>
                                                                <td>Mutation nomenclature according to <a
                                                                        href="https://www.genomic.unimelb.edu.au/mdi/mutnomen/"
                                                                        target="_blank">HGVS</a>
                                                                    standards and using the GenBank NC_000017.10 (hg19
                                                                    assembly)
                                                                    genomic
                                                                    sequence as reference.
                                                                </td>

                                                            </tr>

                                                            <tr>

                                                                <td><a id="g_description_GRCh38"></a>g_description_GRCh38
                                                                </td>
                                                                <td>Mutation nomenclature according to <a
                                                                        href="https://www.genomic.unimelb.edu.au/mdi/mutnomen/"
                                                                        target="_blank">HGVS</a>
                                                                    standards and using the GenBank NC_000017.11 (hg38
                                                                    assembly)
                                                                    genomic
                                                                    sequence as reference.
                                                                </td>

                                                            </tr>

                                                            <tr>
                                                                <td><a id="Type"></a>Type</td>
                                                                <td>Nature of the mutation. The terms
                                                                    occurring in this column are "A:T&gt;C:G" (A to C or
                                                                    T
                                                                    to G base change), "A:T&gt;G:C" (A to G or T to C
                                                                    base
                                                                    change), "A:T&gt;T:A" (A to T or T to A base
                                                                    change),
                                                                    "G:C&gt;A:T" (G to A or C to T base change at non
                                                                    CpG
                                                                    sites), "G:C&gt;A:T at CpG" (G to A or C to T base
                                                                    change at CpG sites),
                                                                    "G:C&gt;C:G" (G to C or C to G base change),
                                                                    "G:C&gt;T:A" (G to T or C to A base change),
                                                                    "tandem" (two consecutive base changes), "ins"
                                                                    (insertion), "del" (deletion) and "complex"
                                                                    (complex changes).
                                                                </td>
                                                            </tr>

                                                            <tr>
                                                                <td><a id="Splice_site"></a>Splice_site</td>
                                                                <td>
                                                                    Annotation on the position of the mutation within
                                                                    conserved
                                                                    nucleotides
                                                                    of
                                                                    <em>p53</em> consensus, criptic or alternative
                                                                    splice sites:
                                                                    <br>
                                                                    <b>consensus SD or SA</b>= the mutation is located
                                                                    at
                                                                    conserved
                                                                    dinucleotides involved in <em>p53</em> consensus
                                                                    splice sites
                                                                    (SD for splice donor site, SA for splice acceptor
                                                                    site)
                                                                    producing
                                                                    the
                                                                    full-lenght <em>p53</em> protein (TA isoform);
                                                                    <br>
                                                                    <b>criptic SD or SA</b>= the mutation is located at
                                                                    conserved
                                                                    dinucleotides
                                                                    involved in splice sites (gt or ag)
                                                                    that have been observed experimentally in
                                                                    <em>p53</em>
                                                                    sequences
                                                                    carrying
                                                                    mutated
                                                                    consensus splice sites;
                                                                    <br>
                                                                    <b>alternative SD or SA</b>= conserved dinucleotides
                                                                    involved in
                                                                    splice
                                                                    sites (gt or ag)
                                                                    responsible for producing <em>p53</em> isoforms beta
                                                                    and
                                                                    gamma;
                                                                    <br>
                                                                    <b>alternative </b>= mutated nucleotides are in the
                                                                    "cassette"
                                                                    sequence
                                                                    responsible for producing the <em>p53</em> delta
                                                                    isoform;
                                                                    <br>
                                                                    <b>no</b>= the position is outside the above
                                                                    mentioned
                                                                    nucleotides.
                                                                    <br>
                                                                    Information on splice site can be found <a
                                                                        target="_blank"
                                                                        href="https://www.web-books.com/MoBio/Free/Ch5A4.htm">here</a>.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="CpG_site"></a>CpG_site</td>
                                                                <td>Yes or No indicate if the position of the mutation
                                                                    falls
                                                                    within a
                                                                    CpG
                                                                    site or not.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="Context3"></a>Context_coding</td>
                                                                <td>Trinucleotide sequence context of variants. The 5'
                                                                    base
                                                                    and
                                                                    3' base
                                                                    of
                                                                    the
                                                                    start position of the variant are indicated on the
                                                                    left
                                                                    and
                                                                    right
                                                                    respectively of the mutated base.
                                                                    This context is provided on the coding strand of the
                                                                    gene
                                                                    sequence
                                                                    and
                                                                    is
                                                                    based on hg38 <em>TP53</em> sequence.
                                                                </td>
                                                            </tr>

                                                            <tr>
                                                                <td><a id="Mut_rate"></a>Mut_rate</td>
                                                                <td>Substitution rates were calculated for all single
                                                                    base
                                                                    substitutions
                                                                    in
                                                                    the
                                                                    coding sequence of <em>p53</em> according to the
                                                                    dinucleotide substitution rates derived from
                                                                    human-mouse
                                                                    aligned
                                                                    sequences
                                                                    of chromosomes 21 and 10
                                                                    (<a href="https://www.ncbi.nlm.nih.gov/pubmed/15262802"
                                                                        target="_blank">Lunter
                                                                        and Hein 2004</a>).
                                                                    The mutation probabilities for a given single
                                                                    nucleotide
                                                                    substitution
                                                                    are
                                                                    calculated by averaging the dinucleotide
                                                                    substitution
                                                                    rates at that position for the forward and reverse
                                                                    strands.
                                                                </td>
                                                            </tr>

                                                            <tr>

                                                                <td><a id="WT_codon"></a>WT_codon</td>
                                                                <td>For mutations in exons, sequence of the codon in
                                                                    which
                                                                    the
                                                                    mutation
                                                                    occurred
                                                                    in NM_000546.5 transcript.
                                                                </td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="Mutant_codon"></a>Mutant_codon</td>
                                                                <td>Base sequence of the mutated codon in NM_000546.5
                                                                    transcript.
                                                                </td>

                                                            </tr>
                                                            <tr>
                                                                <td><a id="WT_AA"></a>WT_AA</td>
                                                                <td>Wild-type amino acid encoded at the codon in which
                                                                    the mutation occurred (three-letter amino acid
                                                                    abbreviation). Check <a target="_blank"
                                                                                            href="aa_codes">AA letter
                                                                        code</a> and
                                                                    <a target="_blank" href="genetic_code">Genetic
                                                                        code</a>
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="Mutant_AA"></a>Mutant_AA</td>
                                                                <td>Mutated amino acid encoded at the
                                                                    codon in which the mutation occurred (three-letter
                                                                    amino acid abbreviation). The chain terminating
                                                                    mutations due
                                                                    to single base substitutions are designated by
                                                                    "stop". &nbsp;&nbsp;
                                                                    Check <a target="_blank" href="aa_codes">AA letter
                                                                        code</a>
                                                                    and
                                                                    <a target="_blank" href="genetic_code">Genetic
                                                                        code</a>
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="ProtDescription"></a>ProtDescription</td>
                                                                <td>Mutation description at the protein level as
                                                                    recommended
                                                                    by
                                                                    <a
                                                                            href="resources#mutation">HGVS</a> and using
                                                                    the
                                                                    Uniprot
                                                                    reference
                                                                    sequence P04637.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="Mut_rateAA"></a>Mut_rateAA</td>
                                                                <td>Mutation rate of amino-acid substitution calculated
                                                                    by
                                                                    summing up
                                                                    the
                                                                    nucleotide substitution
                                                                    rates. This value is only valid for amino-acid
                                                                    substitutions
                                                                    resulting from single nucleotide substitutions.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="Effect"></a>Effect</td>
                                                                <td>Effect of the mutation. The terms
                                                                    occurring in this column are: missense (change of
                                                                    one
                                                                    amino-acid),
                                                                    nonsense
                                                                    (introduction of a stop codon), FS
                                                                    (frameshift), silent (no change in the protein
                                                                    sequence),
                                                                    splice (mutations located in the two first and two
                                                                    last
                                                                    conserved
                                                                    nucleotides of the introns and are thus predicted to
                                                                    alter
                                                                    splicing,
                                                                    or mutations that have been shown to alter splicing
                                                                    experimentally),
                                                                    other
                                                                    (inframe deletions or insertions), intronic
                                                                    (mutations
                                                                    in
                                                                    introns
                                                                    outside
                                                                    splicing sites), NA (mutations upstream in 5' or 3'
                                                                    UTR).

                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="Polymorphism"></a>Polymorphism</td>
                                                                <td>Polymorphic status of the gene variation. <br>
                                                                    <b> Validated </b>: MAF &gt; 0.001 in ESP6500, 1000G
                                                                    or
                                                                    gnomAD
                                                                    databases;
                                                                    <br>
                                                                    <b> No </b>: not reported or reported at MAF &lt;
                                                                    0.001
                                                                    in
                                                                    ESP6500,
                                                                    1000G,
                                                                    or gnomAD databases;<br>
                                                                    <b> NA </b>: not applicable.
                                                                </td>
                                                            </tr>
                                                            <tr>

                                                                <td><a id="SNPlink"></a>SNPlink</td>
                                                                <td>Link to NCBI SNP database.</td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="gnomADlink"></a>gnomADlink</td>
                                                                <td>Link to gnomAD database.</td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="SourceDatabases"></a>SourceDatabases</td>
                                                                <td>SNP databases from which the variants have been
                                                                    extracted.
                                                                </td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="PubMedlink"></a>PubMedlink</td>
                                                                <td>PubMedID of the publications in which was reported
                                                                    the
                                                                    polymorphic
                                                                    status of
                                                                    the variant.
                                                                </td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="DomainFunction"></a>Domain function</td>
                                                                <td>Function of the domain in which the mutated residue
                                                                    is
                                                                    located.
                                                                </td>

                                                            </tr>
                                                            <tr>
                                                                <td><a id="ResidueFunction"></a>Residue function</td>
                                                                <td>Known function of the wild-type residue. When the
                                                                    function
                                                                    is not
                                                                    known
                                                                    but
                                                                    the structure
                                                                    is known, the solvent accessibility (SA) of the
                                                                    residue
                                                                    is
                                                                    indicated
                                                                    by
                                                                    the
                                                                    terms buried, exposed or partially exposed
                                                                    (SA calculated with <a target="_blank"
                                                                                           href="https://wolf.bms.umist.ac.uk/naccess/">Naccess</a>
                                                                    and 1TSR (chain B) structure of <em>p53</em>:
                                                                    &lt;20 = buried, &gt;
                                                                    =20 and
                                                                    &lt;50 = partially exposed, &gt;
                                                                    =50 = exposed).
                                                                </td>
                                                            </tr>
                                                            <tr>

                                                                <td><a id="Hotspot"></a>Hotspot</td>
                                                                <td>"Yes" indicate the a variant is located in a codon
                                                                    defined
                                                                    as a
                                                                    cancer
                                                                    hotspot by <a target="_blank"
                                                                                  href="https://www.ncbi.nlm.nih.gov/pubmed/29247016">Chang
                                                                        (2017)</a>.
                                                                </td>

                                                            </tr>
                                                            <tr>
                                                                <td><a id="StructuralMotif"></a>Structural_motif</td>
                                                                <td>2D and 3D motifs where the mutation is located
                                                                    according
                                                                    to
                                                                    structures
                                                                    described in
                                                                    <a href="https://www.ncbi.nlm.nih.gov/pubmed/8023157"
                                                                       target="_blank">Cho
                                                                        et al. (1994)</a> and <a
                                                                            href="https://www.ncbi.nlm.nih.gov/pubmed/10618702"
                                                                            target="_blank">
                                                                        May and May (1999)</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="SA"></a>SA</td>
                                                                <td>Solvent accessibility of the wild-type residue as
                                                                    calculated
                                                                    with
                                                                    <a target="_blank"
                                                                       href="https://wolf.bms.umist.ac.uk/naccess/">Naccess</a>
                                                                    and the 1TSR (chain B) structure of <em>p53</em>.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="AGVGD_class"></a>AGVGD class</td>
                                                                <td>Missense variant functional predictions by an
                                                                    optimized
                                                                    Align-GVGD
                                                                    tool.
                                                                    Mutations classified as "C0" are considered
                                                                    tolerated
                                                                    while
                                                                    other
                                                                    classes
                                                                    are considered damaging.
                                                                    Further details in <a target="_blank"
                                                                                          href="https://www.ncbi.nlm.nih.gov/pubmed/29775997">Fortuno
                                                                        (2018)</a>.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="BayesDel"></a>BayesDel</td>
                                                                <td>Missense variant functional predictions by BayesDel
                                                                    tool
                                                                    (<a
                                                                            target="_blank"
                                                                            href="https://www.ncbi.nlm.nih.gov/pubmed/27995669">Feng
                                                                        2017)</a> used without allele frequency.
                                                                    <br>Score &gt;=0.16: damaging
                                                                    <br>Score &lt;0.16: tolerated
                                                                    <br>Further details in <a target="_blank"
                                                                                              href="https://www.ncbi.nlm.nih.gov/pubmed/29775997">Fortuno
                                                                        (2018)</a>.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="REVEL"></a>REVEL</td>
                                                                <td>Missense variant functional predictions by REVEL
                                                                    tool
                                                                    (<a
                                                                            target="_blank"
                                                                            href="https://www.ncbi.nlm.nih.gov/pubmed/27666373">Ioannidis
                                                                        2016)</a>.
                                                                    <br>Score &gt;=0.5: damaging
                                                                    <br>Score &lt;0.5: tolerated
                                                                    <br>Further details in <a target="_blank"
                                                                                              href="https://www.ncbi.nlm.nih.gov/pubmed/29775997">Fortuno
                                                                        (2018)</a>.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="SIFT_class"></a>SIFT class</td>
                                                                <td>Functional classification based on <a
                                                                        href="https://sift.jcvi.org/protein_batch_submit.php?species=human"
                                                                        target="_blank">SIFT program</a> using default
                                                                    settings.
                                                                    Missense mutations are classified as "damaging" or
                                                                    "tolerated".
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="Polyphen2"></a>Polyphen2</td>
                                                                <td>Functional classification based on <a
                                                                        href="https://genetics.bwh.harvard.edu/pph2/"
                                                                        target="_blank">Polyphen2</a>
                                                                    HVAR annotations retrieved with <a target="_blank"
                                                                                                       href="https://annovar.openbioinformatics.org/en/latest/">Annovar</a>
                                                                    software:
                                                                    <br>D: probably Damaging
                                                                    <br>P: Possibly damaging
                                                                    <br>B: Benign.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="TA_Ishioka"></a>Transactivation</td>
                                                                <td>Promoter-specific transcriptional activity measured
                                                                    in
                                                                    yeast
                                                                    functional
                                                                    assays and expressed as percent of wild-type
                                                                    activity.
                                                                    Data from <a target="_blank"
                                                                                 href="https://www.ncbi.nlm.nih.gov/pubmed/12826609">Kato
                                                                        (2003)</a></td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="TA_class"></a>TransactivationClass</td>
                                                                <td>Functional classification based on the overall
                                                                    transcriptional
                                                                    activity
                                                                    (TA)
                                                                    on 8 different promoters as measured in yeast assays
                                                                    by
                                                                    <a
                                                                            href="https://www.ncbi.nlm.nih.gov/pubmed/12826609"
                                                                            target="_blank">Kato
                                                                        et al</a>.
                                                                    For each mutant, the median of the 8
                                                                    promoter-specific
                                                                    activities
                                                                    (expressed
                                                                    as percent of the wild-type protein) is calculated
                                                                    and
                                                                    missense
                                                                    mutations
                                                                    are classified as "non-functional" if the median is
                                                                    &lt;=20,
                                                                    "partially functional" if the median is &gt;20 and
                                                                    &lt;=75,
                                                                    "functional"
                                                                    if
                                                                    the median is &gt;75 and &lt;=140, and "supertrans"
                                                                    if
                                                                    the
                                                                    median is
                                                                    &gt;140.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="DNE_LOFclass"></a>DNE_LOFclass</td>
                                                                <td>Functional classification for loss of
                                                                    growth-suppression
                                                                    and
                                                                    dominant-negative activities based on Z-scores from
                                                                    <a target="_blank"
                                                                       href="https://www.ncbi.nlm.nih.gov/pubmed/30224644">Giacomelli
                                                                        et al., (2018)</a> study:
                                                                    <br>DNE_LOF = p53WTNutlin3 Z-score &gt;= 0.61 and
                                                                    Etoposide
                                                                    Z-score
                                                                    &lt;=
                                                                    -0.21;
                                                                    <br>notDNE_notLOF = p53WTNutlin3 Z-score &lt; 0.61
                                                                    and
                                                                    Etoposide
                                                                    Z-score
                                                                    &gt; -0.21;
                                                                    <br>notDNE_LOF if p53WTNutlin3 Z-score &lt; 0.61 and
                                                                    Etoposide
                                                                    Z-score
                                                                    &lt;=
                                                                    -0.21;
                                                                    <br>unclass = others
                                                                </td>
                                                            </tr>

                                                            <tr>
                                                                <td><a id="DNE"></a>DNE class</td>
                                                                <td>Dominant-negative (DN) Effect on transactivation by
                                                                    wild-type <em>p53</em>.
                                                                    <br>
                                                                    Classification established for mutants for which
                                                                    available
                                                                    DN
                                                                    activity
                                                                    on
                                                                    more than
                                                                    2 <em>p53</em>-response elements is available. Data
                                                                    are based
                                                                    on
                                                                    WAF1
                                                                    and RGC
                                                                    promoters in various studies
                                                                    (these promoters were the most frequently used in
                                                                    different
                                                                    studies
                                                                    to
                                                                    assess DNE status), and on
                                                                    two large systematic study (<a target="_blank"
                                                                                                   href="https://www.ncbi.nlm.nih.gov/pubmed/16861262">Dearth
                                                                        et al </a> that includes 76 mutants; <a
                                                                            target="_blank"
                                                                            href="https://www.ncbi.nlm.nih.gov/pubmed/21343334">
                                                                        Monti et al </a> that includes 104 mutants).<br>

                                                                    Mutants were classified as <b>"Yes"</b> if they had
                                                                    dominant-negative
                                                                    activity on both WAF1 and
                                                                    RGC promoters, or on all promoters in the large
                                                                    studies,
                                                                    <b>"Moderate"</b>
                                                                    if they had dominant-negative
                                                                    activity on some but not all promoters, and
                                                                    <b>"No"</b>
                                                                    if
                                                                    they had
                                                                    no
                                                                    dominant-negative
                                                                    activity on both WAF1 and RGC promoters, or none of
                                                                    the
                                                                    promoters in
                                                                    the
                                                                    large studies.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="StructureFunction"></a>Structure/Function
                                                                    class
                                                                </td>
                                                                <td>Functional predictions derived from a computer model
                                                                    that takes into account the 3D structure of WT and
                                                                    mutant proteins and is
                                                                    trained on the transactivaton dataset from
                                                                    <a target="_blank"
                                                                       href="https://www.ncbi.nlm.nih.gov/pubmed/12826609">
                                                                        Kato et al.
                                                                    </a> Mutations are classified as
                                                                    "functional" or "non-functional". More details
                                                                    <a target="_blank" href="dt_method">here</a>.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="EffectGroup3"></a>EffectGroup3</td>
                                                                <td>
                                                                    Mutation classification based on protein 3D
                                                                    structure
                                                                    and
                                                                    mutation
                                                                    type.
                                                                    This classification has been used to derive
                                                                    gentoype-phenotype
                                                                    correlations
                                                                    in sporadic breast cancers (<a target="_blank"
                                                                                                   href="https://www.ncbi.nlm.nih.gov/pubmed/16489069">Olivier
                                                                    et al., 2006</a>).
                                                                    <br>
                                                                    <b>0</b>=silent+intron;
                                                                    <br>
                                                                    <b>1</b>=missense in DNA-binding
                                                                    loops(L2,H1,L3,L1,S2,S2',H2);
                                                                    <br>
                                                                    <b>2</b>=other missense;
                                                                    <br>
                                                                    <b>3</b>=inFrame del/ins;
                                                                    <br>
                                                                    <b>4</b>=FS+splice+nonsense.
                                                                </td>
                                                            </tr>
                                                            <tr>

                                                                <td><a id="SwissProtLink"></a>SwissProtLink</td>
                                                                <td>SwissProt identification number with link to the
                                                                    variant
                                                                    page of the
                                                                    SwissProt database.
                                                                </td>

                                                            </tr>

                                                            <tr>

                                                                <td><a id="SomaticCount"></a>Somatic_count</td>
                                                                <td>Number of occurence in the IARC somatic dataset
                                                                    (number
                                                                    of
                                                                    tumors
                                                                    reported
                                                                    to carry this somatic mutation). Total count is
                                                                    29891 in
                                                                    R20.
                                                                </td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="GermlineCount"></a>Germline_count</td>
                                                                <td>Number of occurence in the IARC germline dataset
                                                                    (number
                                                                    of
                                                                    pedigree/individual carriers of this germline
                                                                    mutation).
                                                                    Total count
                                                                    is
                                                                    1532
                                                                    in R20.
                                                                </td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="CellLineCount"></a>CellLine_count</td>
                                                                <td>Number of occurence in the IARC cell-line dataset
                                                                    (number of
                                                                    cell-lines
                                                                    reported to carry this mutation). Total cell-line
                                                                    count
                                                                    is
                                                                    2766 in
                                                                    R20.
                                                                </td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="COSMIClink"></a>COSMIClink</td>
                                                                <td>Link to mutation ID in COSMIC database.</td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="CLINVARlink"></a>CLINVARlink</td>
                                                                <td>Link to ClinVar database.</td>

                                                            </tr>
                                                            <tr>

                                                                <td><a id="tcga_icgc_genie_count"></a>TCGA_ICGC_GENIE_count
                                                                </td>
                                                                <td>Sum of mutation occurence from TCGA (MC3), ICGC
                                                                    (v28)
                                                                    and
                                                                    GENIE (V5)
                                                                    datasets. Total count is 23570.
                                                                </td>

                                                            </tr>


                                                            </tbody>
                                                        </table>
                                                    </div>

                                                    <br>

                                                    <h4>Predicted effect on splicing:</h4>
                                                    <div class="pb-5">
                                                        <table class="table">
                                                            <thead>
                                                            <tr>
                                                                <th>Column head</th>
                                                                <th>Description</th>
                                                            </tr>
                                                            </thead>
                                                            <tbody>
                                                            <tr>
                                                                <td><a id="SiteType"></a>Site Type</td>
                                                                <td>Indicate if the predicted splice site is an acceptor
                                                                    site or donor site.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="p53Site"></a><em>p53</em> Site</td>
                                                                <td>Indicate if the predicted splice site correspond to
                                                                    a canonical <em>p53</em> splice site.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="WTscore"></a>WT score</td>
                                                                <td>Fit score of the predicted splice site for the
                                                                    non-mutated sequence (scores
                                                                    are specific of prediction tools).
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="MUTscore"></a>MUT score</td>
                                                                <td>Fit score of the predicted splice site for the
                                                                    mutated sequence (scores are specific of prediction
                                                                    tools).
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="SpliceVariation"></a>Variation</td>
                                                                <td>Predicted effect of the mutation on the predicted
                                                                    splice site.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="SpliceSource"></a>Source</td>
                                                                <td>Prediction tool used.</td>
                                                            </tr>
                                                            </tbody>
                                                        </table>
                                                    </div>

                                                    <br>

                                                    <h4>Predicted effect on <em>p53</em> protein isoforms:</h4>

                                                    <br>
                                                    <p>The predictions provided are based on whether the mutation falls
                                                        within the specific
                                                        isoform. For a description of <em>p53</em> isoforms, see <a
                                                                href="p53isoforms"
                                                                target="_blank">here</a>.
                                                    </p>
                                                    <div class="pb-5">
                                                        <table class="table">
                                                            <thead>
                                                            <tr>
                                                                <th>Column head</th>
                                                                <th>Description</th>
                                                            </tr>
                                                            </thead>
                                                            <tbody>
                                                            <tr>
                                                                <td><a id="MainIsoforms"></a>TAp53alpha</td>
                                                                <td>Indicate if the mutation fall within the canonical
                                                                    isoform coding for the full length <em>p53</em>
                                                                    protein.
                                                                </td>
                                                            </tr>
                                                            <tr>
                                                                <td><a id="OtherIsoforms"></a>TAp53beta....deltap53alpha
                                                                </td>
                                                                <td>Indicate if the mutation fall within the specified
                                                                    isoform.
                                                                </td>
                                                            </tr>
                                                            </tbody>
                                                        </table>
                                                    </div>
                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapseSM" aria-expanded="false"
                                                        aria-controls="collapseSM">
                                                </button>
                                                Somatic Mutations Found in Human Tumor Samples
                                            </div>
                                            <div class="collapse" id="collapseSM">
                                                <div class="bg-light p-3 my-3">
                                                    <h4>somaticMutationDataIARC <em>TP53</em> Database, R20.txt</h4>

                                                    <p> This dataset contains <em>TP53</em> somatic mutations identified
                                                        in
                                                        human tumor samples (including metastasis and cell-lines).
                                                        It includes data on the type and position of mutations,
                                                        detailed information on the tumor in which the mutations
                                                        have been found,
                                                        and on various characteristics of the patients in which the
                                                        tumor developed. </p>

                                                    <p> Each row in the downloaded tab-delimited text file
                                                        represents a single mutation reported in a tumor sample with
                                                        an arbitrarily assigned unique identification number.
                                                        A unique identification number is also attributed to the
                                                        tumor sample and to the patient. Table content is as
                                                        follows:
                                                    </p>

                                                    <table class="table">
                                                        <tbody>
                                                        <tr valign="middle">
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        <tr>
                                                            <td colspan="2" align="left"><i>The first set of columns
                                                                describe the mutation.</i></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Mutation_ID</td>
                                                            <td>Unique identification number for a Sample/Mutation
                                                                association.<br>
                                                                Tandem mutations (two adjacent base substitutions)
                                                                are considered as one mutation event;
                                                                therefore tandem mutations have only one
                                                                identification number and are a single record.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>MUT_ID....SwissProtLink</td>
                                                            <td><a class="jump-link" href="#MUT_id">see mutation
                                                                annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td colspan="2" align="left"><i>The second set of
                                                                columns are assigned to the description of the organ
                                                                site, tissue and type of
                                                                lesion in which the mutation has been identified.
                                                                The descriptions given in the publication are
                                                                translated into the standards
                                                                of the International Classification of Diseases for
                                                                Oncology (ICD-O 3rd Edition, World Health
                                                                Organization, Geneva, 2000)
                                                                and <a href="https://www.snomed.org/"
                                                                       target="_blank">SNOMED</a>.<br>
                                                                For information on tumor classification, grading and
                                                                staging, check out
                                                                <a href="https://training.seer.cancer.gov/icdo3/"
                                                                   target="_blank">ICD-O training at SEER</a>,
                                                                <a href="https://www.cancer.gov/cancertopics"
                                                                   target="_blank">Cancer
                                                                    Information at NCI</a> and <a
                                                                        href="https://www.oncologychannel.com/"
                                                                        target="_blank">Oncologychannel.com</a>.</i>
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Sample_name"></a>Sample_Name
                                                            </td>
                                                            <td>A sample name is assigned as follows: first 3
                                                                letters of the first author's name, year of
                                                                publication (2 digit), followed by the ID number
                                                                indicated in the publication.
                                                                The same name or number can occur several times as
                                                                in some samples more than one mutation has been
                                                                reported.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Sample_ID</td>
                                                            <td>Unique sample identification number. This number
                                                                allows the automatic retrieval of samples with
                                                                multiple mutations.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Sample_source...TNM</td>
                                                            <td><a class="jump-link"
                                                                   href="#Sample_source">see sample annotations</a>
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="p53_IHC"></a>p53_IHC</td>
                                                            <td><em>p53</em> immunostaining graded as ‘positive’,
                                                                ‘negative’
                                                                or ‘+/-‘. ND stands for not done.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Add_Info</td>
                                                            <td>Any relevant additional information is entered
                                                                here.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td colspan="2" align="left">
                                                                <i>The third set of columns are assigned to the
                                                                    description of the patient origin and
                                                                    life-style.
                                                                    They contain heterogeneous notes, usually
                                                                    comments emphasized by authors reporting the
                                                                    mutations.
                                                                    It should be noted that this information is
                                                                    generally qualitative. No quantitative
                                                                    information on exposure of risk
                                                                    factors is included in the database. This
                                                                    information does not presuppose that a formal,
                                                                    causal link has been established between
                                                                    such factors and the mutation described.
                                                                    Moreover, for most exogenous risk factors,
                                                                    individual exposure has not been
                                                                    monitored. <b> This information is given solely
                                                                        to (i) permit the retrieval of mutations
                                                                        found in patients
                                                                        belonging to defined groups or having
                                                                        specific risk factors, and (ii) facilitate
                                                                        access to the corresponding publications.
                                                                        For
                                                                        detailed comparison between exposure groups,
                                                                        users are invited to perform their own
                                                                        analysis based on the information given in
                                                                        the
                                                                        original publication.</b>
                                                                </i>
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Individual_ID</td>
                                                            <td>Unique identification number for an individual
                                                                included in the database. It is automatically
                                                                assigned
                                                                by the database system.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Sex...Country</td>
                                                            <td><a class="jump-link" href="#Sex">see
                                                                patient annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="TP53polymorphism"></a>TP53polymorphism
                                                            </td>
                                                            <td>Presence of a polymorphism in <em>TP53</em> gene.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Germline_mutation"></a>Germline_mutation
                                                            </td>
                                                            <td>Germline mutation detected in any gene in the
                                                                patient.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Family_history"></a>Family_history
                                                            </td>
                                                            <td>Information on the presence or absence of cancers in
                                                                the family of the patient.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Tobacco"></a>Tobacco</td>
                                                            <td>Information on the smoking status of the patient.
                                                                Terms occurring in this column are 'smoker' (with
                                                                qualitative amount in brackets), 'non-smoker',
                                                                'passive-smoker' and 'chewer'.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Alcohol"></a>Alcohol</td>
                                                            <td>Information on the drinking status of the patient.
                                                                Terms occurring in this column are 'drinker'
                                                                (with qualitative amount in brackets), and
                                                                'non-drinker'.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Exposure"></a>Exposure</td>
                                                            <td>Risk factors to which the patient has been exposed
                                                                to, such as aflatoxins, radon, thorotrast, etc...
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Infectious_agent"></a>Infectious_agent
                                                            </td>
                                                            <td>Pathogen (virus or bacteria) detected in the
                                                                patient.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Ref_ID</td>
                                                            <td>Unique identification number for the reference in
                                                                which the mutation is described.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>PubMed</td>
                                                            <td>PubMed reference number provided by NCBI.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Exclude_analysis</td>
                                                            <td>Studies that we recommend to exclude from any
                                                                analysis because of dubious quality.
                                                                Such studies are identified based on the following
                                                                criteria: they report several samples with
                                                                multiple mutations, and/or a high proportion of rare
                                                                variants or variants classified as functional.

                                                            </td>
                                                        </tr>
                                                        </tbody>
                                                    </table>


                                                    <br>

                                                    <h4>somaticMutationReference-IARC <em>TP53</em> Database, R20.txt
                                                    </h4>

                                                    <p>This file lists the publications in which are described the
                                                        mutations
                                                        and gives the method used to detect the mutations. Each row
                                                        (record) represents a
                                                        citation with an arbitrarily assigned unique identification
                                                        number
                                                        (Ref_ID). See <a
                                                                class="jump-link"
                                                                href="#Referencedescription">standardized
                                                            annotations</a> for the
                                                        description of the column content.
                                                    </p>


                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapsePrev" aria-expanded="false"
                                                        aria-controls="collapsePrev">
                                                </button>
                                                Prevalence of <em>TP53</em> Somatic Mutations by Tumor Site
                                            </div>
                                            <div class="collapse" id="collapsePrev">
                                                <div class="bg-light p-3 my-3">
                                                    <h4>prevalenceSomaticIARC <em>TP53</em> Database, R20.txt</h4>

                                                    <p> This dataset contains information on the proportion of
                                                        tumors that carry a somatic <em>TP53</em>
                                                        mutation extracted from publications contained in the
                                                        Somatic dataset, and in additional publications that do not
                                                        give a
                                                        detailed description of the mutations (many studies do not
                                                        provide detailed information on each mutation detected
                                                        but rather report their results in the form of summary
                                                        tables, preventing their inclusion in the somaticMutation
                                                        dataset), or publications reporting negative results (no
                                                        mutation found, thus not included in the somaticMutation
                                                        dataset).
                                                    </p>

                                                    <p>For each study, the total number of tumors or tissue samples
                                                        analyzed, and the number of these samples which were found
                                                        to contain
                                                        a mutation is provided.The reference paper, method of
                                                        mutation detection and country of origin of the patients are
                                                        also indicated.
                                                        When the same research team published several papers that
                                                        describe the same set of samples, data from the most recent
                                                        or more complete paper are used.
                                                    </p>

                                                    <table class="table">
                                                        <thead>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        </thead>
                                                        <tbody>
                                                        <tr>
                                                            <td>Prevalence_ID</td>
                                                            <td>Unique entry identification number.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Topography...Morpho_code</td>
                                                            <td><a class="jump-link"
                                                                   href="#Topography">see sample annotations</a>
                                                            </td>
                                                        </tr>


                                                        <tr>
                                                            <td>Sample_analyzed</td>
                                                            <td>Number of tumor samples analyzed for <em>TP53</em>
                                                                mutations.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Sample_mutated</td>
                                                            <td>Number of tumor samples with a mutation in <em>TP53</em>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Country...Development</td>
                                                            <td><a class="jump-link"
                                                                   href="#Country">see patient annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Comment</td>
                                                            <td>Any relevant information.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Ref_ID...PubMed</td>
                                                            <td><a class="jump-link"
                                                                   href="#Referencedescription">see reference
                                                                annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Tissue_processing....exon 11</td>
                                                            <td><a class="jump-link"
                                                                   href="#TissueProcessing">see method
                                                                annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Exclude_Analysis</td>
                                                            <td>Studies that we recommend to exclude from any
                                                                analysis because of data quality issues.
                                                                Studies are labeled as 'exclude' if: they report
                                                                several samples with multiple mutations
                                                                in patients with no specific genetic background or
                                                                exposure to mutagen;
                                                                they report more variants that are classified as
                                                                functional or partially functional (based on TA
                                                                class) than variants classified as non-functional;
                                                                mutations are not precisely described and can not be
                                                                fully annotated in the database;
                                                                several mutations in the series are reported with
                                                                errors (such as position and base that do not fit,
                                                                or report of neutral polymorphisms as somatic
                                                                mutations).

                                                            </td>
                                                        </tr>
                                                        </tbody>
                                                    </table>


                                                </div>

                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapsePrevLiver" aria-expanded="false"
                                                        aria-controls="collapsePrevLiver">
                                                </button>
                                                Prevalence of the R249S <em>TP53</em> Mutations in Liver Cancer
                                            </div>

                                            <div class="collapse" id="collapsePrevLiver">
                                                <div class="bg-light p-3 my-3">


                                                    <p> This dataset contains data on the prevalence of the c.747G&gt;T
                                                        (p.R249S) mutation in liver cancers.
                                                        It includes studies that have screened at least exon 7 of <em>TP53</em>
                                                        by sequencing, and studies that have searched for this specific
                                                        mutation by RFLP.
                                                        The presence of this mutation in hepatocellular carcinomas has
                                                        been linked to exposure to aflatoxins and HBV, and may thus
                                                        constitutes a
                                                        biomarker of exposure. This dataset has been released with the
                                                        R15 version of the database and has not been updtaed since then.
                                                    </p>

                                                    <p> The file <label
                                                            class="strong">R249Sprevalence_IARC_R15.txt </label> is a
                                                        tab-delimited text file, that contains the following info:
                                                    </p>

                                                    <table class="table">
                                                        <thead>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        </thead>
                                                        <tbody>
                                                        <tr>
                                                            <td>Ref_ID...PubMed</td>
                                                            <td><a class="jump-link"
                                                                   href="#Referencedescription">see reference
                                                                annotations</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Country</td>
                                                            <td><a class="jump-link"
                                                                   href="#Country">see
                                                                patient annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Sample_analyzed</td>
                                                            <td>Number of tumor samples analyzed for the c.747G&gt;T
                                                                (p.R249S) <em>TP53</em> mutation.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Count_R249S</td>
                                                            <td>Number of tumor samples containing the c.747G&gt;T
                                                                (p.R249S) <em>TP53</em> mutation.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Remark</td>
                                                            <td>Any relevant information.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Method</td>
                                                            <td>Comment on method if different from sequencing.</td>
                                                        </tr>

                                                        </tbody>
                                                    </table>


                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapsePV" aria-expanded="false"
                                                        aria-controls="collapsePV">
                                                </button>
                                                Prognostic Value of <em>TP53</em> Somatic Mutations
                                            </div>
                                            <div class="collapse" id="collapsePV">
                                                <div class="bg-light p-3 my-3">


                                                    <h4>prognosisSomatic-IARC <em>TP53</em> Database,R20</h4>

                                                    <p>This dataset includes information on all studies that have
                                                        analyzed the
                                                        relationship between <em>p53</em> mutations and cancer
                                                        prognosis. For
                                                        each study,
                                                        the patient cohort, study settings and a summary of the results
                                                        are
                                                        described. When the same research team published several papers
                                                        with
                                                        increasing number of patients, the most recent paper with the
                                                        largest
                                                        dataset is used.
                                                    </p>

                                                    <p>Many of these studies do not provide detailed information on each
                                                        mutation detected but rather report their results in the form of
                                                        summary
                                                        tables. Such publications have been included in the prognosis
                                                        dataset but not in the somaticMutation dataset.
                                                        For some of them, the mutations have been published in a
                                                        previous paper and can be retrieved with the
                                                        Cross_Ref_ID study identifier (see below).
                                                    </p>

                                                    <p>The downloaded file contains the following information:</p>

                                                    <table class="table">
                                                        <thead>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        </thead>
                                                        <tbody>
                                                        <tr>
                                                            <td>Prognosis_ID</td>
                                                            <td>Unique entry identification number.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Topography</td>
                                                            <td><a class="jump-link"
                                                                   href="#Topography">see sample annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Morphology</td>
                                                            <td><a class="jump-link"
                                                                   href="#Morphology">see sample annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Population</td>
                                                            <td><a class="jump-link"
                                                                   href="#Population">see patient annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Country</td>
                                                            <td><a class="jump-link"
                                                                   href="#Country">see
                                                                patient annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Institution</td>
                                                            <td>Name of the hospital(s) where the patients have been
                                                                recruited.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Period</td>
                                                            <td>Time period (year) during which the patients have been
                                                                recruited.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Inclusion criteria</td>
                                                            <td><a href="https://training.seer.cancer.gov/icdo3/"
                                                                   target="_blank">ICD-O</a>
                                                                (3rd edition) or SNOMED code for morphology
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Treatment</td>
                                                            <td>Treatment protocol used for most of the patients.<br>
                                                                SU, surgery; CX, chemotherapy; RX, radiotherapy; pre-op,
                                                                pre-operative; CP, cyclophosphamide; CISP, cisplatin;
                                                                doxo, doxorubicin; 5-FU, 5-fluorouracil
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Median FU</td>
                                                            <td>Median follow-up time of the patients in month.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Range FU</td>
                                                            <td>Range follow-up time of the patients in month.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Cohort</td>
                                                            <td>Number of patients/tumors analyzed for <em>TP53</em>
                                                                mutations.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><em>p53</em> mutations</td>
                                                            <td>Number of patients/tumors with a mutation in
                                                                <em>TP53</em>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Percent mutated</td>
                                                            <td>Proportion of mutated tumors (%).</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Parameter_analyzed</td>
                                                            <td>Clinical parameter analyzed (patient survival and/or
                                                                tumor response to treatment).
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Association</td>
                                                            <td>Summary result: association with the presence of a <em>TP53</em>
                                                                mutation.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Result</td>
                                                            <td>Main findings.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Ref_ID...PubMed</td>
                                                            <td><a class="jump-link"
                                                                   href="#Referencedescription">see
                                                                annotations</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Exclude_analysis</td>
                                                            <td>Papers that we recommend to exclude from analysis
                                                                because of dubious data quality (report several samples
                                                                with multiple mutations, and/or a high proportion of
                                                                rare variants or variants classified as neutral or
                                                                functional).
                                                            </td>
                                                        </tr>

                                                        </tbody>
                                                    </table>
                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapseGM" aria-expanded="false"
                                                        aria-controls="collapseGM">
                                                </button>
                                                Germline Mutations in LFS/LFL Families
                                            </div>
                                            <div class="collapse" id="collapseGM">
                                                <div class="bg-light p-3 my-3">


                                                    <h4>germlineMutationData-IARC <em>TP53</em> Database,R20</h4>

                                                    <p>Inherited <em>TP53</em> mutations are associated with a rare
                                                        autosomal
                                                        dominant disorder,
                                                        the <a target="_blank" href="refs_corner#lifraumeni">Li-Fraumeni
                                                            syndrome (LFS)</a>.
                                                    </p>

                                                    <p>This dataset contains information on individuals that are
                                                        carriers of a <em>TP53</em> germline mutation and families in
                                                        which at
                                                        least one family member
                                                        has been identied as a carrier of a germline mutation in the
                                                        <em>TP53</em> gene. <strong>Criteria for inclusion are the
                                                            following:
                                                            a) individuals carrying a sequenced <em>TP53</em> germline
                                                            mutation,
                                                            affected or not by a cancer, b) individuals affected by a
                                                            cancer and
                                                            belonging to a family in which at least one family member
                                                            has been identified as a carrier of a germline mutation in
                                                            the <em>TP53</em> gene.</strong>
                                                    </p>

                                                    <p> Each row (record) in the downloaded file represents a tumor
                                                        found in an individual having a <em>TP53</em>
                                                        germline mutation. The file contains the following information:
                                                    </p>

                                                    <table class="table">

                                                        <tbody>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        </tbody>
                                                        <tbody>
                                                        <tr>
                                                            <td>Family_ID</td>
                                                            <td>Unique family identification number.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Family_code"></a>Family_Code</td>
                                                            <td>Name or number given in the original
                                                                publication or an arbitrarily-assigned name, usually the
                                                                3 first
                                                                letters of the first author's name and the publication
                                                                date.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Country</td>
                                                            <td><a class="jump-link"
                                                                   href="#Country">see
                                                                annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Family_class"></a>Class</td>
                                                            <td>Family classification:<br>
                                                                <strong>LFS</strong> = strict clinical definition of
                                                                Li-Fraumeni
                                                                syndrome (defined by Li and Fraumeni as a Proband with
                                                                sarcoma
                                                                &lt;45 years with a first degree relative with cancer at
                                                                &lt;45
                                                                and another first/second degree relative with cancer at
                                                                &lt;45 or
                                                                sarcoma at any age);
                                                                <br>
                                                                <strong>LFL</strong> = Li-Fraumeni like for the extended
                                                                clinical
                                                                definition of Li-Fraumeni (including Birch definition:
                                                                proband with any childhood cancer or sarcoma, brain
                                                                tumor or
                                                                adrenocortical carcinoma at &lt;45 years, with one first
                                                                or second
                                                                degree relative with sarcoma, breast cancer, brain
                                                                tumor,
                                                                leukemia, or adrenocortical carcinoma at any age, plus
                                                                one first
                                                                or second degree relative in the same lineage with any
                                                                cancer
                                                                diagnosed under age 60; Eeles definition E1: two
                                                                different tumors
                                                                which are part of extended LFS in first or second degree
                                                                relatives
                                                                at any age (sarcoma, breast cancer, brain tumor,
                                                                leukemia,
                                                                adrenocortical tumor, melanoma, prostate cancer,
                                                                pancreatic
                                                                cancer); Eeles definition E2: sarcoma at any age in the
                                                                proband
                                                                with two of the following (two of the tumors may be in
                                                                the same
                                                                individual): breast cancer at &lt;50 years and/or brain
                                                                tumor,
                                                                leukemia, adrenocortical tumor, melanoma, prostate
                                                                cancer,
                                                                pancreatic cancer at &lt;60 years or sarcoma at any
                                                                age).
                                                                <br>
                                                                <strong>FH</strong>: family history of cancer which does
                                                                not
                                                                fulfil LFS or any of the LFL definitions (Birch, Eeles
                                                                E1 or E2,);
                                                                <br>
                                                                <strong>No FH</strong>: no family history of cancer.;
                                                                FH= Family
                                                                history of cancer (not fulfilling the definition of
                                                                LFS/LFL); No=
                                                                no family history of cancer; ?= unknown.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Generations_analyzed"></a>Generations_analyzed
                                                            </td>
                                                            <td> Number of generations analyzed in the family.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Germline_mutation_g"></a>Germline_mutation
                                                            </td>
                                                            <td>A <em>TP53</em> germline mutation has been identified.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>MUT_ID...TAclass</td>
                                                            <td><a class="jump-link"
                                                                   href="#ExonIntron">see mutation annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Individual_ID</td>
                                                            <td>Unique identification number for an individual included
                                                                in the database.
                                                                It is automatically assigned by the database system.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Individual_code"></a>Individual_code
                                                            </td>
                                                            <td>Code or number given in the original publication or an
                                                                arbitrarily-asigned code,
                                                                usually the family code followed by the position of the
                                                                individual in the family tree.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="FamilyCase"></a>FamilyCase</td>
                                                            <td>Family case in the pedigree, such as proband (index
                                                                case), mother, father...
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="FamilyCase_group"></a>FamilyCase_group
                                                            </td>
                                                            <td>Degree of relationship to the proband.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Sex</td>
                                                            <td>Gender of the individual.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Germline_carrier"></a>Germline_carrier
                                                            </td>
                                                            <td>
                                                                <em>TP53</em> mutation status of the individual: <b>confirmed</b>=
                                                                the individual has been tested for the presence of the
                                                                mutation and the
                                                                mutation has been found; <b>obligatory</b>= the
                                                                individual has not been tested for the presence of the
                                                                mutation but must be carrier based on
                                                                the mutation status of the other individuals in the
                                                                pedigree; <b>50%prob.</b>= there is a chance of 50% that
                                                                the individual is a mutation carrier;
                                                                <b>negative</b>= the individual has been tested for the
                                                                presence of the mutation and the mutation has not been
                                                                found; <b>NA</b>=
                                                                the individual has not been tested for the presence of
                                                                the mutation.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Mode_of_inheritance"></a>Mode_of_inheritance
                                                            </td>
                                                            <td>Mode of mutation inheritance: P=paternal, M=maternal, M&amp;P=maternal
                                                                and paternal, de novo= mutation that has not been
                                                                inherited, "de novo, mosaic"= mutation that has not been
                                                                inherited and is present in a subpopulation of cells,
                                                                na=not known.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Dead"></a>Dead</td>
                                                            <td>Living status of the individual at
                                                                time of follow-up. 0=alive; 1=dead
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Unaffected"></a>Unaffected</td>
                                                            <td>Disease status of the individual at time of follow-up. 0
                                                                = affected by cancer; 1 = not affected by
                                                                cancer.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Age</td>
                                                            <td>Age of the individual at the time of follow-up.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>Topography</td>
                                                            <td><a class="jump-link"
                                                                   href="#Topography">see annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Morphology</td>
                                                            <td><a class="jump-link"
                                                                   href="#Morphology">see annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>Age at diagnosis</td>
                                                            <td>Age of the individual at the time of diagnosis of the
                                                                tumor.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Ref_ID</td>
                                                            <td>Reference number indicating the
                                                                publication in which the mutation is described. This
                                                                number
                                                                corresponds to the Ref_ID number in the GermlineRefR20
                                                                file.
                                                            </td>
                                                        </tr>

                                                        </tbody>
                                                    </table>

                                                    <br>
                                                    <h4>germlineMutationReference-IARC <em>TP53</em> Database,R20</h4>

                                                    <p> Each row represents a reference identified by a unique
                                                        identification
                                                        number (Ref_ID). See <a class="jump-link"
                                                                                href="#Referencedescription">standard
                                                            annotations</a>
                                                        for a description of the column content.
                                                    </p>

                                                    <br>
                                                    <h4>germlinePrevalenceIARC <em>TP53</em> Database, R20</h4>

                                                    <p> This dataset includes studies reporting <em>TP53</em> germline
                                                        mutation
                                                        screening in large cohorts of patients selected based on various
                                                        criteria (family history of cancer, specific cancer diagnosis,
                                                        ...)
                                                        Each row represents the result of the analysis of <em>TP53</em>
                                                        germline
                                                        mutation status in a selected cohort.
                                                    </p>

                                                    <table class="p53table">

                                                        <tbody>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Diagnosis"></a>Diagnosis</td>
                                                            <td>Tumor site or clinical description of the selected
                                                                cohort.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Cohort"></a>Cohort</td>
                                                            <td>Detailed criteria for patient selection.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Cases_analyzed"></a>Cases analyzed
                                                            </td>
                                                            <td>Number of patients included in the mutation screen.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Cases_mutated"></a>Cases mutated
                                                            </td>
                                                            <td>Number of patients found to carry a <em>TP53</em>
                                                                mutation.
                                                                Details on mutations can be found in the dataset of
                                                                germline mutations when the information was provided,
                                                                but many studies
                                                                do not provide a detailed list of mutations.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="MutationPrevalence"></a>Mutation
                                                                prevalence
                                                            </td>
                                                            <td>Percent of mutated cases.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="GermPrevRemark"></a>Remark</td>
                                                            <td>Any further information on the cohort or method.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>PubMed</td>
                                                            <td>PubMed ID with link to ncbi database.</td>
                                                        </tr>

                                                        </tbody>
                                                    </table>

                                                    <br>
                                                    <h4>germlineFrequency_IARCTP53Database_R20</h4>

                                                    <p> This new dataset includes studies reporting the frequency of
                                                        individual <em>TP53</em> germline mutations in case-control
                                                        series using
                                                        NGS for the screening of the entire coding regions of
                                                        <em>TP53</em>.
                                                    </p>

                                                    <table class="p53table">

                                                        <tbody>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        <tr>
                                                            <td>MUT_ID....DNEclass</td>
                                                            <td><a class="jump-link"
                                                                   href="#MUT_id">see
                                                                mutation annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Freq_cases"></a>Freq_cases</td>
                                                            <td>Frequency of the <em>TP53</em> variant in cases.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Freq_controls"></a>Freq_controls
                                                            </td>
                                                            <td>Frequency of the <em>TP53</em> variant in controls.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Total_Cases"></a>Total_Cases</td>
                                                            <td>Number of patients included in the "case" group.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="N_Cases"></a>N_Cases</td>
                                                            <td>Number of cases found to carry the specific
                                                                <em>TP53</em>
                                                                variant.
                                                                Details on mutation phenotype may be found in the
                                                                dataset of germline mutations if enough details is
                                                                provided on patients and tumor type.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Total_Controls"></a>Total_Controls
                                                            </td>
                                                            <td>Number of patients included in the "control" group.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="N_Controls"></a>N_Controls</td>
                                                            <td>Number of controls found to carry the specific
                                                                <em>TP53</em>
                                                                variant.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>DataSource</td>
                                                            <td>PubMed ID of the paper from which data have been
                                                                extracted.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="StudyDetails"></a>StudyDetails</td>
                                                            <td>Description of the case and control groups.</td>
                                                        </tr>

                                                        </tbody>
                                                    </table>


                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapseMM" aria-expanded="false"
                                                        aria-controls="collapseMM">
                                                </button>
                                                Functional Activities of Missense Mutations
                                            </div>
                                            <div class="collapse" id="collapseMM">
                                                <div class="bg-light p-3 my-3">
                                                    <p> Data on the biological properties of <em>p53</em> mutant
                                                        proteins in
                                                        functional assays performed in yeast or human cells, are
                                                        provided in two datasets.</p>

                                                    <h4>functionalAssessment-IARC <em>TP53</em> Database, R20</h4>
                                                    <a id="FunctionalAssessment"></a>

                                                    <table class="table" style="border-top: 1px solid #dee2e6;">
                                                        <tbody>
                                                        <tr>
                                                            <td><i class="fas fa-exclamation-triangle"></i></td>
                                                            <td>
                                                                In this dataset, data were extracted from publications
                                                                that report functional assessment of <em>p53</em> mutant
                                                                proteins
                                                                in human or yeast cells, assessed either by transfection
                                                                and overexpression of mutant proteins,
                                                                or by assessment of endogenous mutants. Comparison
                                                                between mutants requires caution since
                                                                functional assays differ from one study to the other, in
                                                                particular with respect to
                                                                the expression vector (which influences the level of
                                                                expression of the mutant protein),
                                                                the <em>p53</em>-responsive elements (generic consensus
                                                                sequence
                                                                versus gene-specific
                                                                response elements from WAF1, BAX or PIG3), and the
                                                                recipient cells that have been used.
                                                            </td>
                                                        </tr>
                                                        </tbody>
                                                    </table>
                                                    <div class="py-3">
                                                        <p>The functional properties of mutant proteins that are
                                                            included in
                                                            this dataset are:</p>

                                                        <ul>
                                                            <li>transcriptional activities on various well-described
                                                                <em>p53</em>-RE
                                                            </li>
                                                            <li>dominant negative effects on the activities of wild-type
                                                                <em>p53</em>
                                                            </li>
                                                            <li>capacity to induce apoptosis, cell-cycle arrest or
                                                                checkpoints
                                                                in human cells
                                                            </li>
                                                            <li>capacity to transactivate promoters that are not induced
                                                                by
                                                                wild-type <em>p53</em>
                                                            </li>
                                                            <li>ability to promote cell growth and confer
                                                                tumorigenicity
                                                            </li>
                                                            <li>sensitivity to temperature changes regarding their
                                                                ability to
                                                                transactivate specific promoters
                                                            </li>
                                                        </ul>
                                                    </div>

                                                    <div class="py-3">
                                                        The functional results have been organised in 5 columns for (1)
                                                        conserved
                                                        wild-type properties, (2) complete or partial loss of wild-type
                                                        properties, (3) dominant-negative effects, (4) gain
                                                        of function and (5) temperature sensitivity. The cell system is
                                                        indicated in two columns and a detailed reference to the
                                                        published report is given.
                                                    </div>

                                                    <table class="table">

                                                        <thead>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        </thead>
                                                        <tbody>
                                                        <tr>
                                                            <td>Function_ID</td>
                                                            <td>Unique identification number for each entry.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>ProtDescription...Structural_motif</td>
                                                            <td><a class="jump-link" href="#ProtDescription">see
                                                                mutation
                                                                annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Codon72AA"></a>Codon 72</td>
                                                            <td>Amino-acid at codon 72 of <em>p53</em> (<a
                                                                    target="_blank"
                                                                    href="refs_corner#polymorph">polymorphism</a>)
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Conserved_WT_function"></a>Conserved
                                                                WT function
                                                            </td>
                                                            <td>
                                                                Functional property of mutant that is similar to the
                                                                activity of the wild-type protein.<br>
                                                                <br>
                                                                - Activities of mutant proteins in human or yeast cells:<br>
                                                                <b>DNAb</b> = DNA binding capacity tested by gel-shift
                                                                or ChIP assay;<br>
                                                                <b>TA</b> = transactivation of a reporter gene under the
                                                                control of a <em>p53</em>-response element
                                                                (indicated in brakets, see list <a
                                                                    target="_blank" href="target_genes">here</a>);<br>
                                                                <b>TR</b> = transrepression of a reporter gene under the
                                                                control of a gene-specific response-element
                                                                (name of gene indicated in brakets)<br>
                                                                <b>TETR</b> = capacity to form tetramers;<br>
                                                                <b>x binding</b> = interaction with protein x;<br>
                                                                <br>
                                                                <b>drug sensitivity</b> = conserved capacity to mediate
                                                                cytotoxic effect of drug (specific drug used is
                                                                indicated, <a
                                                                    class="jump-link"
                                                                    href="#List_of_abbreviations">see List of
                                                                abbreviations</a>).<br>
                                                                <br>
                                                                - Activities of over-expressed mutant proteins in human
                                                                cells:<br>
                                                                <b>APO</b> = induction of apoptosis;<br>
                                                                <b>GS</b> = growth suppression measured by colony
                                                                forming assay (CFA), establishment of stable clones, or
                                                                other proliferation assay;<br>
                                                                <b>GA</b> = cell cycle arrest measured by FACS;<br>
                                                                <b>TUMOR-</b> = inhibition of tumorigenicity in nude
                                                                mice;<br>
                                                                <b>up/downregulation</b> = induction or repression of an
                                                                endogenous GENE (in upper-case letters) or protein (in
                                                                lowercase letters);<br>
                                                                <b>HR repression</b> = inhibition of Homologous
                                                                Recombination;<br>
                                                                <br>
                                                                - Biological effect after over-expression in mouse or
                                                                rat embryonic fibroblasts:<br>
                                                                <b>TRANSF-</b> = ability to counteract the
                                                                transformation of primary cells induced by the
                                                                co-transfection of ras or another
                                                                transformant oncogene, such as HPV E7;<br>
                                                                <br>
                                                                "<b>super</b>" indicates that the activity of mutant
                                                                protein is higher than the one of wtp53 (on
                                                                transactivation, induction of apoptosis, DNA binding or
                                                                growth suppression).<br>
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Loss_of_function"></a>Loss of
                                                                Function
                                                            </td>
                                                            <td>WTp53 functional property that is lost by the mutant
                                                                protein.<br>
                                                                Same annotations as in previous column, with
                                                                <b>partial</b>" indicating that the loss of function is
                                                                partial (residual activity).
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Dominant_negative_activity"></a>Dominant
                                                                negative activity
                                                            </td>
                                                            <td>
                                                                Inhibition of the wild-type protein by
                                                                mutant proteins in <strong>transactivation or cell
                                                                growth assays</strong>.<br>
                                                                - <b>Yes</b> = the mutant protein counteract the
                                                                activity of the wild-type
                                                                protein when the two proteins are co-expressed in human
                                                                or yeast
                                                                cells (the <em>p53</em>-response element or cell growth
                                                                assay
                                                                performed is indicated in brakets);<br>
                                                                - <b>No</b> = the mutant protein does not counteract the
                                                                effects of the
                                                                wild-type protein.<br>
                                                                <br>
                                                                "<b>moderate</b>" indicates that the mutant protein has
                                                                a partial inhibiting effect on
                                                                the wild-type protein.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Gain_of_function"></a>Gain of
                                                                Function
                                                            </td>
                                                            <td>Functional properties displayed by the
                                                                mutant but not by the wild-type protein.<br>
                                                                <br>
                                                                - Activities of over-expressed mutant proteins in human
                                                                or yeast cells:<br>
                                                                same annotations as in column 9, plus:<br>
                                                                <b>TUMOR+</b> = confer tumorigenic property (in nude
                                                                mice) to transfected cells;<br>
                                                                <b>p73 interference</b> = ability to counteract p73
                                                                activity when both proteins are expressed in a cell
                                                                system;<br>
                                                                <b>Drug resistance</b> = confer resistance to a
                                                                cytotoxic drug (<a class="jump-link"
                                                                                   href="#List_of_abbreviations">see
                                                                    List of
                                                                    abbreviations</a>);<br>
                                                                <b>Growth advantage</b> = increase growth rate. <br>
                                                                <br>
                                                                - Biological effect after over-expression in mouse or
                                                                rat embryonic fibroblasts:<br>
                                                                <b>TRANSF+</b> = ability to cooperate with ras or
                                                                another transformant oncogene, such as HPV E7, in the
                                                                transformation of
                                                                primary cells.<br>
                                                                <br>
                                                                "<b>moderate</b>" indicates that the mutant protein has
                                                                a partial effect on the activity studied;
                                                                "<b>no</b>" indicates that the mutant protein has no
                                                                effect on the activity studied. <br>
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Temperature_sensitivity"></a>Temperature
                                                                sensitivity
                                                            </td>
                                                            <td>Sensitivity of mutant to temperature changes in <strong>transactivation
                                                                assays </strong>
                                                                (the <em>p53</em>-RE is indicated in brackets), and in
                                                                other
                                                                experimental assays (specified in brackets).<br>
                                                                <b>Yes</b> = the activity of the mutant protein is
                                                                affected by the
                                                                temperature at which is preformed the test;<br>
                                                                &nbsp;&nbsp;<b>mut_H</b> = the protein is inactive
                                                                (mutant) at higher temperatures;<br>
                                                                &nbsp;&nbsp;<b>mut_L</b> = the protein is inactive
                                                                (mutant) at lower temperatures;<br>
                                                                <b>No</b> = the activity of the mutant protein is NOT
                                                                affected by the temperature at which is preformed the
                                                                test.<br>
                                                                <br>
                                                                Note that functional tests are performed at different
                                                                temperature in yeast (30ÝC) and human (37ÝC) cells.
                                                                Detailed annotation rules are available <a
                                                                        target="_blank" href="ts_annotations">here</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Reference_Temperature"></a>Temp_ref
                                                            </td>
                                                            <td>Temperature at which experiments have been performed or
                                                                which has been used as reference for temperature
                                                                sensitivity assays.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Cell_assay"></a>Cell assay</td>
                                                            <td>Human = the activity of the mutant
                                                                protein has been tested in human cells.<br>
                                                                Yeast = the activity of the mutant protein has been
                                                                tested in the
                                                                yeast.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Cells"></a>cellLines</td>
                                                            <td>Name of cell-line(s) that have been
                                                                used for testing mutant activities. "<b>(endo)</b>"
                                                                indicates that activities have been tested on endogenous
                                                                mutants.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Assay_design"></a>Assay design</td>
                                                            <td>Indicates if the assay has been performed with or
                                                                without wtp53 as control, or if activity has been tested
                                                                on endogenous mutant.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Method"></a>Method</td>
                                                            <td>Details on type of experimental assay that was performed
                                                                to assess function.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>FRef_ID...PubMed</td>
                                                            <td><a class="jump-link"
                                                                   href="#Referencedescription">see reference
                                                                annotations</a>).
                                                            </td>
                                                        </tr>

                                                        </tbody>
                                                    </table>

                                                    <br>

                                                    <h4>systematicFunctionalAssessment-IARC <em>TP53</em> Database, R20
                                                    </h4>
                                                    <a id="systematicFunctionalAssessment"></a>
                                                    <p>
                                                        The functional data that are included in this dataset were
                                                        provided by Chikashi Ishioka and have been published in
                                                        <a target="_blank"
                                                           href="https://www.ncbi.nlm.nih.gov/pubmed/12826609">Kato et
                                                            al.</a>, <a
                                                            href="https://www.ncbi.nlm.nih.gov/pubmed/15781620"
                                                            target="_blank">Kakudo et al.</a>, and
                                                        <a href="https://www.ncbi.nlm.nih.gov/pubmed/16007150"
                                                           target="_blank">Kawaguchi et al.</a>.
                                                    </p>

                                                    <table class="table">

                                                        <thead>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        </thead>
                                                        <tbody>
                                                        <tr>
                                                            <td>ProtDescription...codon number</td>
                                                            <td><a class="jump-link" href="#ProtDescription">see
                                                                mutation
                                                                annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="TA_Ishioka_Yeast"></a>WAF1nWT,
                                                                MDM2nWT, BAXnWT,...
                                                            </td>
                                                            <td>Promoter-specific transcriptional activity measured in
                                                                yeast functional assays
                                                                and expressed as percent of wild-type activity.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="TA_Ishioka_Saos"></a>WAF1nWT_Saos2,
                                                                MDM2nWT_Saos2,...
                                                            </td>
                                                            <td>Promoter-specific transcriptional activity measured in
                                                                the human cell-line Saos-2. Values are normalized with
                                                                <em>p53</em>-null vector values and expressed as percent
                                                                of
                                                                wild-type activity.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="APO_Ishioka"></a>SubG1nWT_Saos2
                                                            </td>
                                                            <td>Induction of apoptosis by overexpression in Saos-2 cells
                                                                expressed as percent of wild-type activity.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Oligomerisation_yeast</td>
                                                            <td>Capacity of mutant protein to form oligomer:<br>
                                                                TETR=can form tetramer, <br>
                                                                DIM=can form dimer but not tetramer, <br>
                                                                MON= can not oligomerarize.
                                                            </td>
                                                        </tr>

                                                        </tbody>
                                                    </table>
                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapseHC" aria-expanded="false"
                                                        aria-controls="collapseHC">
                                                </button>
                                                <em>TP53</em> status of Human Cell-Lines
                                            </div>
                                            <div class="collapse" id="collapseHC">
                                                <div class="bg-light p-3 my-3">


                                                    <h4>CellLinesMutationStatus-IARC <em>TP53</em> Database,R20</h4>

                                                    <p>This dataset includes cell-lines that have been screened for
                                                        <em>TP53</em> mutation and have been
                                                        published in the scientific literature, or in the
                                                        <a href="https://www.sanger.ac.uk/genetics/CGP/CellLines/"
                                                           target="_blank">Sanger cell-line database</a> or the
                                                        <a href="https://www.broadinstitute.org/ccle/home"
                                                           target="_blank">Broad Cancer cell-line Encyclopedia</a>.
                                                    </p>

                                                    <table class="table">

                                                        <thead>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        </thead>
                                                        <tbody>
                                                        <tr>
                                                            <td>Sample_ID</td>
                                                            <td>Unique sample identification number.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="CellName"></a>Sample_name</td>
                                                            <td>Name of the cell-line.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="ATCC"></a>ATCC_ID</td>
                                                            <td>Identification number of the <a
                                                                    href="https://www.lgcpromochem-atcc.com/common/catalog/cellBiology/cellBiologyIndex.cfm"
                                                                    target="_blank">ATCC database</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Cosmic"></a>Cosmic_ID</td>
                                                            <td>Link to sample cell-line data in the <a
                                                                    href="https://www.sanger.ac.uk/genetics/CGP/CellLines/"
                                                                    target="_blank">Cancer Cell Line Project</a>
                                                                of
                                                                <a href="https://www.sanger.ac.uk/genetics/CGP/cosmic/"
                                                                   target="_blank">COSMIC</a> databases of the
                                                                Sanger Institute.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="depmap"></a>depmap_ID</td>
                                                            <td>Link to sample cell-line data in the <a
                                                                    href="https://depmap.org/portal/"
                                                                    target="_blank">depmap project</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Short_topo...Tumor_origin</td>
                                                            <td><a class="jump-link"
                                                                   href="#Topography">see sample annotations</a>
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Add_info</td>
                                                            <td></td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="CellSex"></a>Sex</td>
                                                            <td>Gender of the patient from whom the cell-line has
                                                                been isolated.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="CellAge">Age</a></td>
                                                            <td>Age at cancer diagnosis of the patient from whom the
                                                                cell-line has been isolated.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Country...Population</td>
                                                            <td><a class="jump-link"
                                                                   href="#Country">see
                                                                patient annotations</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Germline_mutation</td>
                                                            <td>Germline mutation in <em>TP53</em> or any other gene
                                                                carried
                                                                by the individual from which the cell-line has been
                                                                isolated.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Infectious_agent</td>
                                                            <td>Infectious agent (virus or bacteria) detected in the
                                                                individual from which the cell-line has been
                                                                isolated.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Tobacco</td>
                                                            <td>Smoking habit of the individual from which the
                                                                cell-line has been isolated.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Alcohol</td>
                                                            <td>Drinking habit of the individual from which the
                                                                cell-line has been isolated.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Exposure</td>
                                                            <td>Reported exposure of the individual from which the
                                                                cell-line has been isolated.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>KRAS_status</td>
                                                            <td>Status of KRAS gene. WT=
                                                                wild-type; MUT=mutant (base change indicated in
                                                                brackets)
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Other_mutations</td>
                                                            <td>Name of other genes in which a mutation has been
                                                                identified.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="TP53_status"></a>TP53status
                                                            </td>
                                                            <td>Status of <em>TP53</em> gene. WT= wild-type gene
                                                                sequence;
                                                                MUT= mutated gene sequence;
                                                                NULL= entire gene deletion;
                                                                LOE= loss of gene expression without gene mutation.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>p53_IHC</td>
                                                            <td><em>p53</em> immuno-staining status.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>p53_LOH</td>
                                                            <td>Loss of heterozygocity at <em>p53</em> locus. Yes= LOH,
                                                                No=
                                                                no LOH, NI= non informative, NA= no information
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>MUT_ID...</td>
                                                            <td>
                                                                <em>TP53</em> mutation description and functional
                                                                properties,
                                                                <a class="jump-link"
                                                                   href="#Mutationdescription">see
                                                                    mutation and function annotations</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Ref_ID...</td>
                                                            <td> Same as somatic Ref_ID, <a class="jump-link"
                                                                                            href="#Referencedescription">see
                                                                reference annotations</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Tissue_processing...</td>
                                                            <td><a class="jump-link"
                                                                   href="#Detectionmethod">see method
                                                                annotations</a>.
                                                            </td>
                                                        </tr>

                                                        </tbody>
                                                    </table>
                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapseMouse" aria-expanded="false"
                                                        aria-controls="collapseMouse">
                                                </button>
                                                Mouse Models with Engineered TP3
                                            </div>
                                            <div class="collapse" id="collapseMouse">
                                                <div class="bg-light p-3 my-3">

                                                    <p>The dataset contains mouse models with engineered <em>p53</em>
                                                        that
                                                        are compiled in the <a
                                                                href="https://cancermodels.nci.nih.gov/camod/login.do"
                                                                target="_blank">
                                                            caMOD database</a> or reported in the scientific
                                                        literature.
                                                        Data curated at caMOD were courteously provided by the caMOD
                                                        team. A direct link to the caMOD web site is available for
                                                        each model for a detailed
                                                        description of model genetics and phenotypes. Data reported
                                                        in the literature but not compiled in caMOD are curated at
                                                        IARC and a link to PubMed
                                                        abstract is provided. For a detailed description of model
                                                        genetics and phenotypes, please refer to caMOD and/or
                                                        original publication
                                                    </p>
                                                    <br>
                                                    <h4>Mouse ModelsIARC <em>TP53</em> Database, R20</h4>
                                                    <br>
                                                    <table class="table">

                                                        <tbody>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="ModelDescriptor"></a>Model descriptor</td>
                                                            <td>Model name as indicated in caMOD or original
                                                                publication.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="AffectedOrgans"></a>Affected organs</td>
                                                            <td>List of organs affected or targeted by transgene.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="AAchangeHuman"></a>AA change in human</td>
                                                            <td>Amino-acid substitution. Note that amino-acids are
                                                                numbered according to the human sequence.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="caMOD"></a>caMOD link</td>
                                                            <td>Model ID from caMOD database.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="PubMed"></a>PubMed</td>
                                                            <td>PMID link to original publication.</td>
                                                        </tr>

                                                        </tbody>
                                                    </table>
                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapseEIM" aria-expanded="false"
                                                        aria-controls="collapseEIM">
                                                </button>
                                                Experimentally Induced Mutations
                                            </div>
                                            <div class="collapse" id="collapseEIM">
                                                <div class="bg-light p-3 my-3">

                                                    <p>This dataset contains list of mutations in the human
                                                        <em>TP53</em> gene
                                                        obtained from
                                                        mutagenicity assays in the <a
                                                                href="https://www.ncbi.nlm.nih.gov/pubmed/16777987"
                                                                target="_blank">Hupki mouse model</a> (MEF cells treated
                                                        with the indicated carcinogen agent) or in
                                                        a yeast assay. See original papers for detailed methods.
                                                    </p>

                                                    <br>
                                                    <h4>inducedMutationIARC <em>TP53</em> Database, R20</h4>
                                                    <br>
                                                    <table class="table">
                                                        <thead>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        </thead>
                                                        <tbody>
                                                        <tr>
                                                            <td>MUT_ID</td>
                                                            <td>Unique ID for the mutation, used across datasets.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="ExpIndMutExposure"></a>Exposure</td>
                                                            <td>Agents to which were exposed the cells.</td>
                                                        </tr>
                                                        <tr>
                                                            <td>c_description</td>
                                                            <td>Mutation described on the cDNA sequence. <a
                                                                    class="jump-link"
                                                                    href="#MUT_id">See
                                                                mutation annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td>g_description</td>
                                                            <td>Mutation described at the genome level. <a
                                                                    class="jump-link"
                                                                    href="#MUT_id">See
                                                                mutation annotations</a></td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="ExpIndMutModel"></a>Model</td>
                                                            <td>Experimental assay/model used.</td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="ExpIndMutCloneID"></a>Clone_ID</td>
                                                            <td>ID of cell clone isolated from the exposed cell
                                                                population.
                                                            </td>
                                                        </tr>

                                                        <tr>
                                                            <td><a id="ExpIndMutAddInfo"></a>Add info</td>
                                                            <td>Additional details provided on assay or cell clone as
                                                                derived from original publication.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>PubMed</td>
                                                            <td>PMID with link to PubMed abstract that describe the
                                                                model.
                                                            </td>
                                                        </tr>

                                                        </tbody>
                                                    </table>


                                                </div>
                                            </div>
                                        </li>
                                    </ul>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="accordion-item">
                        <h2 class="accordion-header" id="headingSA">
                            <button class="accordion-button" type="button" data-bs-toggle="collapse"
                                    data-bs-target="#collapseSA" aria-expanded="true" aria-controls="collapseSA">
                                Standardized Annotations
                            </button>
                        </h2>
                        <div id="collapseSA" class="accordion-collapse collapse show" aria-labelledby="headingSA"
                             data-bs-parent="#accordionUM">
                            <div class="accordion-body">
                                <div>
                                    <ul class="list-group list-group-flush">
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapseSamp" aria-expanded="false"
                                                        aria-controls="collapseSamp">
                                                </button>
                                                Sample
                                            </div>
                                            <div class="collapse" id="collapseSamp">
                                                <div class="bg-light p-3 my-3">

                                                    <p>Tumor samples are classified according to standards of the
                                                        International Classification of Diseases
                                                        for Oncology (ICD-O 3rd Edition, World Health Organization,
                                                        Geneva, 2000) and <a href="https://www.snomed.org/"
                                                                             target="_blank">SNOMED</a>.</p>

                                                    <p> For information on tumor classification, grading and
                                                        staging, check out
                                                        <a href="https://training.seer.cancer.gov/icdo3/"
                                                           target="_blank">
                                                            ICD-O training at SEER</a>, <a
                                                                href="https://www.cancerstaging.org/cstage/index.html"
                                                                target="_blank">
                                                            Collaborative staging initiative</a>, <a
                                                                href="https://www.cancer.gov/cancertopics/"
                                                                target="_blank">
                                                            Cancer Information at NCI</a>, and <a
                                                                href="https://www.oncologychannel.com/"
                                                                target="_blank">Oncologychannel.com</a>.
                                                    </p>
                                                    <table class="table">
                                                        <thead>
                                                        <tr>
                                                            <th>Column head</th>
                                                            <th>Description</th>
                                                        </tr>
                                                        </thead>
                                                        <tbody>
                                                        <tr>
                                                            <td><a id="Sample_source"></a>Sample_source</td>
                                                            <td>Nature of the sample from which the
                                                                mutation has been identified: cell-line, surgery
                                                                (surgical or
                                                                autopsy specimen, including fresh samples and
                                                                archival, pathology
                                                                specimen), biopsy, xenograft, body fluid (blood,
                                                                saliva,
                                                                urine...).
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Tumor_origin"></a>Tumor_Origin</td>
                                                            <td>Origin of the tumor sample. Terms
                                                                occurring in this column are: primary, secondary
                                                                (second primary
                                                                tumor in the same patient), metastasis (with the
                                                                localisation of
                                                                the metastasis in brackets), recurrent (tumor
                                                                recurrence).
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Topography"></a>Topography</td>
                                                            <td>Site of the tumor defined by organ or
                                                                group of organs, according to the ICD-O
                                                                nomenclature. (examples:
                                                                "colon", "brain", "bronchus and
                                                                lung"). Note that some tumors are annoted
                                                                "Head&amp;Neck,NOS" or "Colorectum,NOS"
                                                                because no detail is given in the original
                                                                publication (NOS= not
                                                                otherwise specified).<br>

                                                                For the database search tool, a short name is used
                                                                in place of the
                                                                ICD-O name (example: "Lung" for "bronchus and
                                                                lung"). <a target="_blank" href="topo">See
                                                                    a
                                                                    numerical list of topographies</a>.<br>
                                                                <strong>For metastasis, the topography corresponds
                                                                    to the primary
                                                                    site of the tumor and the site of metastasis is
                                                                    indicated in
                                                                    brackets in the tumor_origin field.</strong>
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Short_topo"></a>Short_topography</td>
                                                            <td>For the database search tool, a short
                                                                name is used in place of the ICD-O name (example:
                                                                "Lung"
                                                                for "bronchus and lung"). <a target="_blank"
                                                                                             href="topo">See
                                                                    a numerical list of topographies</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Topo_code"></a>Topo_code</td>
                                                            <td><a target="_blank" href="topo">ICD-O
                                                                code</a> for topography.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="SubTopography"></a>Sub_topography</td>
                                                            <td>Precise identification of anatomic
                                                                site, organ or tissue. The description given in the
                                                                publication is
                                                                translated to ICD-O nomenclature.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Morphology"></a>Morphology</td>
                                                            <td>
                                                                Tumor type, including morphology and/or histologic type.
                                                                The terminology used is based on ICD-O (2nd and 3rd
                                                                editions) and SNOMED classifications. Terms have been
                                                                added, such as 'normal tissue' or 'na'. <a
                                                                    target="_blank" href="morph"> See alphabetical
                                                                list of morphologies.
                                                            </a>
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="MorphoCode"></a>Morpho_code</td>
                                                            <td>ICD-O or SNOMED codes for morphology.</td>
                                                        </tr>
                                                        <tr>

                                                            <td><a id="Grade"></a>Grade</td>
                                                            <td>Information on tumor grade, as given in the cited
                                                                publication.
                                                            </td>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="Stage"></a>Stage</td>
                                                            <td>Information on tumor stage, as given in the cited
                                                                publication.
                                                            </td>

                                                        </tr>
                                                        <tr>
                                                            <td><a id="TNM"></a>TNM</td>
                                                            <td>TNM classification (Tumor size, Node status, Metastasis
                                                                status) for staging. For information on this
                                                                classification system, click <a target="_blank"
                                                                                                href="tnm_class">here</a>.
                                                            </td>
                                                        </tr>
                                                        </tbody>
                                                    </table>
                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapsePA" aria-expanded="false"
                                                        aria-controls="collapsePA">
                                                </button>
                                                Patient
                                            </div>
                                            <div class="collapse" id="collapsePA">
                                                <div class="bg-light p-3 my-3">
                                                    <table class="table">

                                                        <tbody>
                                                        <tr>

                                                            <th>Column head</th>
                                                            <th>Description</th>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="Sex"></a>Sex</td>
                                                            <td>Sex of the patient (M for male, F for female).</td>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="Age"></a>Age</td>
                                                            <td>Age of the patient at the time of
                                                                diagnosis.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Ethnicity"></a>Ethnicity</td>
                                                            <td>Ethnicity of the patient (when
                                                                available). Groups are defined as: Asian, Black,
                                                                Caucasian...
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Country"></a>Country</td>
                                                            <td>Country/Region in which the patient was living at the
                                                                time of surgery.
                                                                When not otherwise specified in the original
                                                                publication, the country corresponding to the address of
                                                                the hospital is entered.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Population"></a>Population</td>
                                                            <td>Grouping by population. To see the country/population
                                                                classification <a
                                                                        href="country">click
                                                                    here</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Region"></a>Region</td>
                                                            <td>Grouping by region. To see the country/region
                                                                classification <a
                                                                        href="country">click
                                                                    here</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Development"></a>Development</td>
                                                            <td>Grouping by development status. To see the
                                                                country/development classification <a
                                                                        href="country">click
                                                                    here</a>.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Geo_area"></a>Geo_area</td>
                                                            <td>City or region within the country of
                                                                living of the patient. When not specified in the
                                                                original
                                                                publication, the city where the surgery has been done is
                                                                entered.
                                                            </td>
                                                        </tr>

                                                        </tbody>
                                                    </table>
                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapseRef" aria-expanded="false"
                                                        aria-controls="collapseRef">
                                                </button>
                                                Reference
                                            </div>
                                            <div class="collapse" id="collapseRef">
                                                <div class="bg-light p-3 my-3">


                                                    <p>The same references (same Ref_ID) are used for the somatic,
                                                        prevalence and prognosis data sets. Independent references
                                                        are used for
                                                        the Function and Germline data sets.
                                                    </p>

                                                    <table class="table">

                                                        <tbody>
                                                        <tr>

                                                            <th>Column head</th>
                                                            <th>Description</th>

                                                        </tr>
                                                        <tr>
                                                            <td><a id="Ref_ID"></a>Ref_ID</td>
                                                            <td>Unique identification number for a reference.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td>Cross_Ref_ID</td>
                                                            <td>Ref_ID of a reference containing related data or
                                                                additional information.
                                                            </td>
                                                        </tr>
                                                        <tr>

                                                            <td><a id="Title"></a>Title</td>
                                                            <td>Title of the publication.</td>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="Author"></a>Authors</td>
                                                            <td>List of authors.</td>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="Year"></a>Year</td>
                                                            <td>Year of publication.</td>

                                                        </tr>
                                                        <tr>
                                                            <td><a id="Journal"></a>Journal</td>
                                                            <td>Name of the journal (<a
                                                                    href="https://www.ncbi.nlm.nih.gov/nlmcatalog/journals"
                                                                    target="_blank">PubMed catalogue</a>)
                                                            </td>
                                                        </tr>
                                                        <tr>

                                                            <td><a id="Volume"></a>Volume</td>
                                                            <td>Volume number.</td>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="Start_page"></a>Start_page</td>
                                                            <td>First page number.</td>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="End_page"></a>End_page</td>
                                                            <td>Last page of article.</td>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="PubMed"></a>PubMed_entry</td>
                                                            <td>PubMed identification number from <a
                                                                    href="https://www.ncbi.nlm.nih.gov/pubmed/"
                                                                    target="_blank">NCBI</a>.
                                                            </td>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="Comment"></a>Comment</td>
                                                            <td>Any relevant information</td>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="Exclude_analysis"></a>Exclude_analysis
                                                            </td>
                                                            <td>Papers that we recommend to exclude from analysis
                                                                because of dubious data quality (report several
                                                                samples with multiple mutations, and/or a high
                                                                proportion of rare variants or variants classified
                                                                as neutral or functional).
                                                            </td>

                                                        </tr>
                                                        <tr>

                                                            <td><a id="WGS_WXS"></a>WGS_WXS</td>
                                                            <td>Whole genome or whole exome sequencing study.</td>

                                                        </tr>

                                                        </tbody>
                                                    </table>


                                                </div>
                                            </div>
                                        </li>
                                        <li class="list-group-item">
                                            <div>
                                                <button class="btn-view-details"
                                                        data-bs-toggle="collapse"
                                                        data-bs-target="#collapseMDM" aria-expanded="false"
                                                        aria-controls="collapseMDM">
                                                </button>
                                                Mutation Detection Method
                                            </div>
                                            <div class="collapse" id="collapseMDM">
                                                <div class="bg-light p-3 my-3">
                                                    <table class="table">
                                                        <tbody>
                                                        <tr>

                                                            <th>Column head</th>
                                                            <th>Description</th>

                                                        </tr>
                                                        <tr>
                                                            <td><a id="Tissue_processing"></a>Tissue_processing
                                                            </td>
                                                            <td>Indicates if the sample analysed was
                                                                fresh, fixed or frozen.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Start_material"></a>Start_material
                                                            </td>
                                                            <td>Indicates if DNA or RNA was screened for mutations.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Prescreening"></a>Prescreening/Method
                                                            </td>
                                                            <td>Prescreening method used to select sample to be
                                                                sequenced:
                                                                ‘SSCP’ for single strand polymorphism, ‘DGE’ for
                                                                denaturant gel electrophoresis, ‘FASAY’ for yeast assay,
                                                                ‘none’ if no prescreening was done, etc…
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Material_sequenced"></a>Material_sequenced
                                                            </td>
                                                            <td>Indicates if the DNA or RNA was cloned or not (direct)
                                                                before sequencing.
                                                            </td>
                                                        </tr>
                                                        <tr>
                                                            <td><a id="Exons_analyzed"></a>Exon2-11</td>
                                                            <td>Exons that have been screened for mutation. In the
                                                                downloaded file,
                                                                "-1" or "TRUE" indicate that the exon has been screened
                                                                and "0" or "FALSE" indicate that it has not been
                                                                screened.
                                                            </td>
                                                        </tr>
                                                        </tbody>
                                                    </table>
                                                </div>
                                            </div>
                                        </li>
                                    </ul>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="accordion-item">
                        <h2 class="accordion-header" id="headingLA">
                            <button class="accordion-button" type="button" data-bs-toggle="collapse"
                                    data-bs-target="#collapseLA" aria-expanded="true" aria-controls="collapseLA">
                                List of Abbreviations
                            </button>
                        </h2>
                        <div id="collapseLA" class="accordion-collapse collapse show" aria-labelledby="headingLA"
                             data-bs-parent="#accordionUM">
                            <div class="accordion-body">
                                <table class="table table-sm table-striped">

                                    <tbody>
                                    <tr>

                                        <td>5-aza-CdR</td>
                                        <td>5-aza-2’-deoxycytidine</td>

                                    </tr>
                                    <tr>
                                        <td>AILD</td>
                                        <td>Angioimmunoblastic Lymphadenopathy
                                            with Dysproteinemia
                                        </td>
                                    </tr>
                                    <tr>

                                        <td>ALL</td>
                                        <td>Acute lymphoblastic leukemia</td>

                                    </tr>
                                    <tr>

                                        <td>AML</td>
                                        <td>Acute myeloid leukemia</td>

                                    </tr>
                                    <tr>

                                        <td>APC</td>
                                        <td>Adenomatous polyposis coli gene</td>

                                    </tr>
                                    <tr>

                                        <td>B(a)P</td>
                                        <td>benzo(a)pyrene</td>

                                    </tr>
                                    <tr>

                                        <td>B(a)PDE</td>
                                        <td>benzo(a)pyrene diol epoxide</td>

                                    </tr>
                                    <tr>

                                        <td>BLEO</td>
                                        <td>bleomycin</td>

                                    </tr>
                                    <tr>

                                        <td>CHEM</td>
                                        <td>Chemotherapy</td>

                                    </tr>
                                    <tr>

                                        <td>CIS</td>
                                        <td>Carcinoma <i>in situ</i></td>

                                    </tr>
                                    <tr>

                                        <td>CDDP</td>
                                        <td>cisplatin</td>

                                    </tr>
                                    <tr>

                                        <td>CML</td>
                                        <td>Chronic Myeloid Leukemia</td>

                                    </tr>
                                    <tr>

                                        <td>CMML</td>
                                        <td>Chronic Myelomonocytic Leukemia</td>

                                    </tr>
                                    <tr>

                                        <td>COSMIC</td>
                                        <td>Catalog Of Somatic Mutations in Cancer</td>

                                    </tr>
                                    <tr>

                                        <td>CPT</td>
                                        <td>camptothecin</td>

                                    </tr>
                                    <tr>

                                        <td>ddF</td>
                                        <td>Dideoxyfingerprinting</td>

                                    </tr>
                                    <tr>

                                        <td>DFI</td>
                                        <td>Disease-free interval</td>

                                    </tr>
                                    <tr>

                                        <td>DGE</td>
                                        <td>Denaturant gel electrophoresis</td>

                                    </tr>
                                    <tr>

                                        <td>DNE</td>
                                        <td>Dominant-negative effect</td>

                                    </tr>
                                    <tr>

                                        <td>Dif.</td>
                                        <td>Differentiated</td>

                                    </tr>
                                    <tr>

                                        <td>DBM</td>
                                        <td>DNA-binding motif</td>

                                    </tr>
                                    <tr>

                                        <td>DOX</td>
                                        <td>doxorubicin</td>

                                    </tr>
                                    <tr>

                                        <td>Duke’s</td>
                                        <td>Classification for colon cancer</td>

                                    </tr>
                                    <tr>

                                        <td>EBV</td>
                                        <td>Epstein-Barr Virus</td>

                                    </tr>
                                    <tr>

                                        <td>ER</td>
                                        <td>Estrogen Receptor</td>

                                    </tr>
                                    <tr>

                                        <td>ETO</td>
                                        <td>etoposide</td>

                                    </tr>

                                    <tr>

                                        <td>FAB</td>
                                        <td>FAB classification for ALL</td>

                                    </tr>
                                    <tr>

                                        <td>FAP</td>
                                        <td>Familial adenomatous polyposis</td>

                                    </tr>
                                    <tr>

                                        <td>FGO</td>
                                        <td>Female Genital Organs</td>

                                    </tr>
                                    <tr>

                                        <td>FH</td>
                                        <td>Family history</td>

                                    </tr>
                                    <tr>

                                        <td>FIGO</td>
                                        <td>Classification for gynecological cancers</td>

                                    </tr>
                                    <tr>

                                        <td>GEM</td>
                                        <td>gemcitabine</td>

                                    </tr>
                                    <tr>
                                        <td>H’sD, HD
                                        </td>
                                        <td>Hodgkin’s disease</td>
                                    </tr>
                                    <tr>

                                        <td>HBV</td>
                                        <td>Hepatitis B virus</td>

                                    </tr>
                                    <tr>

                                        <td>HCC</td>
                                        <td>Hepatocellular carcinoma</td>

                                    </tr>
                                    <tr>

                                        <td>HCV</td>
                                        <td>Hepatitis C virus</td>

                                    </tr>
                                    <tr>

                                        <td>HNPCC</td>
                                        <td>Hereditary Nonpolyposis Colorectal Cancer</td>

                                    </tr>
                                    <tr>

                                        <td>HPV</td>
                                        <td>Human papilloma virus</td>

                                    </tr>
                                    <tr>

                                        <td>HNSCC</td>
                                        <td>Head and neck squamous cell carcinoma</td>

                                    </tr>
                                    <tr>

                                        <td>
                                            <a href="https://www.who.int/standards/classifications/other-classifications/international-classification-of-diseases-for-oncology"
                                               target="_blank">ICD-O 3rd</a></td>
                                        <td>International classification of disease for oncology</td>

                                    </tr>
                                    <tr>

                                        <td>IHC</td>
                                        <td>Immunohistochemistry</td>

                                    </tr>
                                    <tr>

                                        <td align="left">KAT</td>
                                        <td align="left">Potassium Antimony Tartrate</td>

                                    </tr>
                                    <tr>

                                        <td align="left">LFL</td>
                                        <td align="left">Li-Fraumeni like syndrome</td>

                                    </tr>
                                    <tr>

                                        <td align="left">LFS</td>
                                        <td align="left">Li-Fraumeni syndrome</td>

                                    </tr>
                                    <tr>

                                        <td align="left">LN</td>
                                        <td align="left">Lymph node</td>

                                    </tr>
                                    <tr>

                                        <td align="left">LOF</td>
                                        <td align="left">Loss of function</td>

                                    </tr>
                                    <tr>

                                        <td align="left">MTX</td>
                                        <td align="left">methotrexate</td>

                                    </tr>
                                    <tr>

                                        <td align="left">MDE</td>
                                        <td align="left">Hydrolink mutation detection enhancement</td>

                                    </tr>
                                    <tr>

                                        <td align="left">MDS</td>
                                        <td align="left">Myelodysplastic syndrome</td>

                                    </tr>
                                    <tr>

                                        <td align="left">MGO</td>
                                        <td align="left">Male Genital Organs</td>

                                    </tr>
                                    <tr>

                                        <td align="left">Moderately dif.</td>
                                        <td align="left">Moderately differentiated</td>

                                    </tr>

                                    <tr>

                                        <td align="left">MMR</td>
                                        <td align="left">MisMatch Repair</td>

                                    </tr>
                                    <tr>

                                        <td align="left">MNNG</td>
                                        <td align="left">N-methyl-N-nitro-N-nitrosoguanidine</td>

                                    </tr>
                                    <tr>

                                        <td align="left">MSI</td>
                                        <td align="left">Microsatellite instable/td&gt;

                                        </td>
                                    </tr>
                                    <tr>

                                        <td align="left">MSS</td>
                                        <td align="left">Microsatellite stable/td&gt;

                                        </td>
                                    </tr>
                                    <tr>

                                        <td align="left">Mx</td>
                                        <td align="left">Presence of distant metastasis according to the TNM
                                            classification
                                        </td>

                                    </tr>
                                    <tr>

                                        <td align="left">NA</td>
                                        <td align="left">not applicable/ not available</td>

                                    </tr>
                                    <tr>

                                        <td align="left">ND</td>
                                        <td align="left">not done</td>

                                    </tr>
                                    <tr>

                                        <td align="left">NDBL</td>
                                        <td align="left">Non-DNA-binding loop</td>

                                    </tr>
                                    <tr>

                                        <td align="left">neg.</td>
                                        <td align="left">negative</td>

                                    </tr>
                                    <tr>

                                        <td align="left">NES</td>
                                        <td align="left">Nuclear Export Signal</td>

                                    </tr>
                                    <tr>

                                        <td align="left">NIM</td>
                                        <td align="left">nimustine</td>

                                    </tr>
                                    <tr>

                                        <td align="left">NLS</td>
                                        <td align="left">Nuclear Localization Signal</td>

                                    </tr>
                                    <tr>

                                        <td align="left">NHL</td>
                                        <td align="left">Non-Hodgkin’s lymphoma</td>

                                    </tr>
                                    <tr>

                                        <td align="left">NOS</td>
                                        <td align="left">Not otherwise specified</td>

                                    </tr>
                                    <tr>

                                        <td align="left">NOS2</td>
                                        <td align="left">Nitric oxide synthase 2</td>

                                    </tr>
                                    <tr>

                                        <td align="left">NSCLC</td>
                                        <td align="left">Non-small cell lung cancer</td>

                                    </tr>
                                    <tr>

                                        <td align="left">Nx</td>
                                        <td align="left">Extent of regional LN metastasis according to the <a
                                                target="_blank"
                                                href="tnm_class">TNM classification</a>
                                        </td>

                                    </tr>
                                    <tr>

                                        <td align="left">OXA</td>
                                        <td align="left">oxaliplatin</td>

                                    </tr>

                                    <tr>
                                        <td align="left">PJS</td>
                                        <td align="left">Peutz-Jeghers Syndrome</td>

                                    </tr>
                                    <tr>

                                        <td align="left">Poorly dif.</td>
                                        <td align="left">Poorly differentiated</td>

                                    </tr>
                                    <tr>

                                        <td align="left">pos.</td>
                                        <td align="left">positive</td>

                                    </tr>
                                    <tr>

                                        <td align="left">PR</td>
                                        <td align="left">Progesterone Receptor</td>

                                    </tr>
                                    <tr>

                                        <td align="left">RA, RAEB, RAS</td>
                                        <td align="left">Classification for Myelodysplastic syndrome</td>

                                    </tr>
                                    <tr>

                                        <td align="left">RAD</td>
                                        <td align="left">Radiotherapy</td>

                                    </tr>
                                    <tr>

                                        <td align="left">RFLP</td>
                                        <td align="left">Restriction fragment length polymorphism</td>

                                    </tr>

                                    <tr>

                                        <td align="left">SCLC</td>
                                        <td align="left">Small cell lung cancer</td>

                                    </tr>
                                    <tr>

                                        <td align="left">SSCP</td>
                                        <td align="left">Single strand conformation polymorphism</td>

                                    </tr>
                                    <tr>

                                        <td align="left">TAX</td>
                                        <td align="left">taxol</td>

                                    </tr>
                                    <tr>

                                        <td align="left">TGGE</td>
                                        <td align="left">Temperature gradient gel electrophoresis</td>

                                    </tr>
                                    <tr>

                                        <td align="left">TOP</td>
                                        <td align="left">topotecan</td>

                                    </tr>
                                    <tr>

                                        <td align="left">Tx</td>
                                        <td align="left">Extent of primary tumor according to
                                            the <a href="tnm_class">TNM classification</a>
                                        </td>
                                    </tr>
                                    <tr>
                                        <td align="left">TxNxMx</td>
                                        <td align="left"><a href="tnm_class">TNM
                                            classification</a> (tumor stage, presence or absence of LN
                                            metastasis, presence or absence of distant metastasis)
                                        </td>

                                    </tr>
                                    <tr>
                                        <td align="left">Undif.</td>
                                        <td align="left">Undifferentiated</td>

                                    </tr>
                                    <tr>

                                        <td align="left">VIN</td>
                                        <td align="left">vinblastine</td>

                                    </tr>
                                    <tr>

                                        <td align="left">VINC</td>
                                        <td align="left">vincristine</td>

                                    </tr>
                                    <tr>

                                        <td align="left">VM26</td>
                                        <td align="left">teniposide</td>

                                    </tr>
                                    <tr>

                                        <td align="left">Well dif.</td>
                                        <td align="left">Well differentiated</td>

                                    </tr>
                                    <tr>

                                        <td align="left">WGS</td>
                                        <td align="left">Whole genome screen</td>

                                    </tr>
                                    <tr>

                                        <td align="left">XP</td>
                                        <td align="left">Xeroderma pigmentosum</td>

                                    </tr>
                                    <tr>

                                        <td align="left">Yeast</td>
                                        <td align="left">Yeast assay</td>

                                    </tr>
                                    <tr>

                                        <td align="left">yo</td>
                                        <td align="left">Years old</td>
                                    </tr>

                                    </tbody>
                                </table>
                            </div>
                        </div>
                    </div>
                    <div class="accordion-item">
                        <h2 class="accordion-header" id="headingGraphs">
                            <button class="accordion-button" type="button" data-bs-toggle="collapse"
                                    data-bs-target="#collapseGraphs" aria-expanded="true"
                                    aria-controls="collapseGraphs">
                                Graphs and Search Options
                            </button>
                        </h2>
                        <div id="collapseGraphs" class="accordion-collapse collapse show" aria-labelledby="headingGraphs"
                             data-bs-parent="#accordionUM">
                            <div class="accordion-body">
                                <ul class="list-group list-group-flush">
                                    <li class="list-group-item">
                                        <div>
                                            <button class="btn-view-details"
                                                    data-bs-toggle="collapse"
                                                    data-bs-target="#collapseSO" aria-expanded="false"
                                                    aria-controls="collapseSO">
                                            </button>
                                            Search Options
                                        </div>
                                        <div class="collapse" id="collapseSO">
                                            <div class="bg-light p-3 my-3">
                                                <ul>
                                                    <li><b>Functional / Structural Data</b>.
                                                        This option allows the functional and structural analysis of
                                                        all
                                                        possible single nucleotide substitutions in <em>TP53</em>
                                                        exonic sequences (including those that have never been
                                                        reported
                                                        in cancer). In addition, all other types of mutations that
                                                        have
                                                        been reported in human samples and validated polymorphisms
                                                        are
                                                        included in this dataset. Functional and structural
                                                        annotations and frequency statistics for these gene
                                                        variations
                                                        can be retrieved with this search option.
                                                        Each dataset entry corresponds to a unique gene variation.
                                                    </li>
                                                    <li><b>Somatic mutations</b>.
                                                        This option allows the retrieval and analysis of <em>TP53</em>
                                                        mutations
                                                        reported as somatic events in tumor samples and cell-lines.
                                                        Each dataset entry corresponds to a mutation identified in a
                                                        human sample.
                                                    </li>
                                                    <li><b>Somatic mutation prevalence</b>.
                                                        This option allows the analysis of the prevalence of
                                                        <em>TP53</em>
                                                        somatic mutations by cancer type and population groups.
                                                        Each dataset entry corresponds to the prevalence of
                                                        <em>TP53</em>
                                                        mutation for a specific type of cancer in a defined human
                                                        population.
                                                    </li>
                                                    <li><b>Germline mutations</b>.
                                                        This option allows the retrieval and analysis of <em>TP53</em>
                                                        mutations
                                                        reported as germline events in human individuals.
                                                        Each dataset entry corresponds to a tumor identified in an
                                                        individual carrier of a <em>TP53</em> germline mutation.
                                                        The searchable dataset only includes cancer-affected
                                                        individuals
                                                        who are confirmed or obligatory carrier of a <em>TP53</em>
                                                        mutation
                                                        (data on non-affected carriers or non-confirmed carrier can
                                                        be
                                                        retrieved by downloading the full dataset with the 'data
                                                        downloads' option).
                                                    </li>
                                                    <li><b>Germline mutations prevalence</b>.
                                                        This table lists diferent studies reporting the prevalence
                                                        of
                                                        <em>TP53</em> germline mutation in selected groups of
                                                        individuals.
                                                    </li>
                                                    <li><b>Cell-lines</b>.
                                                        This option allows the retrieval and analysis of <em>TP53</em>
                                                        mutations
                                                        reported in human cell-lines.
                                                        Each dataset entry corresponds to a mutation identified in a
                                                        cell-line.
                                                    </li>
                                                    <li><b>Mouse models</b>.
                                                        This option allows the display or download of the
                                                        description of
                                                        mouse models with engineered <em>p53</em> that are compiled in
                                                        the caMOD database or reported in the scientific literature.
                                                        Links to caMOD database are available for further details on
                                                        the
                                                        model phenotypes.
                                                    </li>

                                                </ul>
                                            </div>
                                        </div>
                                    </li>
                                    <li class="list-group-item">
                                        <div>
                                            <button class="btn-view-details"
                                                    data-bs-toggle="collapse"
                                                    data-bs-target="#collapseMDG" aria-expanded="false"
                                                    aria-controls="collapseMDG">
                                            </button>
                                            Mutation Distribution Graphs
                                        </div>
                                        <div class="collapse" id="collapseMDG">
                                            <div class="bg-light p-3 my-3">
                                                <ul>
                                                    <li><b>Mutation type</b>. Proportion of mutations classified by
                                                        their
                                                        nature (base change, insertions, deletions....):
                                                        number of mutations of each class divided by the total
                                                        number of
                                                        mutations selected (% is shown).
                                                    </li>
                                                    <li><b>Codon distribution</b>. Proportion of exonic point
                                                        mutations
                                                        at
                                                        each codon position:
                                                        number of mutations at each codon position divided by the
                                                        total
                                                        number of exonic mutations selected (% is shown).
                                                    </li>
                                                    <li><b>Exon/intron distribution</b>. Proportion of mutations in
                                                        each
                                                        exon/intron:
                                                        number of mutations within each Exon/intron divided by the
                                                        total
                                                        number of mutations selected (% is shown).
                                                    </li>
                                                    <li><b>3D JMOL graph</b>. Residues (within the central domain of
                                                        <em>p53</em>
                                                        protein -codons 96 to 289) are highlighted according to the
                                                        proportion of exonic mutations at this position (start site
                                                        of
                                                        mutation) among all selected mutations:
                                                        number of mutations at each codon position divided by the
                                                        total
                                                        number of exonic mutations selected: red colored are the
                                                        most
                                                        frequently mutated, yellow colored the less frequently
                                                        mutated,
                                                        orange are intermediate.
                                                    </li>
                                                    <li><b>Mutation effect</b>. Proportion of mutations classified
                                                        according
                                                        to their predicted effect on protein sequence (missense,
                                                        nonsense,
                                                        frameshift ins/del, …):
                                                        number of mutations of each class divided by the total
                                                        number of
                                                        mutations selected (% is shown).
                                                    </li>
                                                    <li><b>Point mutation</b>. Proportion of single amino-acid
                                                        substitutions
                                                        classified according to their predicted effect on protein
                                                        sequence
                                                        (missense, nonsense, silent):
                                                        number of mutations of each class divided by the total
                                                        number of
                                                        point mutations selected (% is shown).
                                                    </li>
                                                    <li><b>Point mutation dot-plot</b>. Each dot represent a
                                                        specific
                                                        point
                                                        mutation, colored according to their predicted effect on
                                                        protein
                                                        sequence (missense in blue, nonsense in red and silent in
                                                        green);
                                                        the x axis shows the proportion of the specific mutation in
                                                        the
                                                        selected dataset (% of total point mutations in the selected
                                                        dataset); the Y axis shows the predicted mutation rate for
                                                        the
                                                        particular point mutation (see mutation annotations).
                                                    </li>
                                                    <li><b>SIFT</b>. Proportion of missense mutations classified
                                                        according
                                                        to their predicted deleterious/damaging or neutral/tolerated
                                                        effect
                                                        based on SIFT algorithm:
                                                        number of mutations of each class divided by the total
                                                        number of
                                                        missense mutations selected (% is shown).
                                                    </li>
                                                    <li><b>SIFT dot-plot</b>. Each dot represent a specific point
                                                        mutation,
                                                        colored according to their predicted deleterious/damaging or
                                                        neutral/tolerated effect based on SIFT algorithm;
                                                        the x axis shows the proportion of the specific mutation in
                                                        the
                                                        selected dataset (% of total point mutations in the selected
                                                        dataset); the Y axis shows the predicted mutation rate for
                                                        the
                                                        particular point mutation (see mutation annotations).
                                                    </li>
                                                    <li><b>Transactivation</b>. Proportion of missense mutations
                                                        classified
                                                        according to their experimentally measured transactivation
                                                        activities (based on FASAY):
                                                        number of mutations of each class divided by the total
                                                        number of
                                                        missense mutations selected (% is shown).
                                                    </li>
                                                    <li><b>Transactivation dot-plot</b>. Each dot represent a
                                                        specific
                                                        point
                                                        mutation, colored according to their experimentally measured
                                                        transactivation activities;
                                                        the x axis shows the proportion of the specific mutation in
                                                        the
                                                        selected dataset (% of total point mutations in the selected
                                                        dataset); the Y axis shows the predicted mutation rate for
                                                        the
                                                        particular point mutation (see mutation annotations).
                                                    </li>
                                                </ul>
                                            </div>
                                        </div>
                                    </li>
                                    <li class="list-group-item">
                                        <div>
                                            <button class="btn-view-details"
                                                    data-bs-toggle="collapse"
                                                    data-bs-target="#collapseTDG" aria-expanded="false"
                                                    aria-controls="collapseTDG">
                                            </button>
                                            Tumor Distribution Graphs
                                        </div>
                                        <div class="collapse" id="collapseTDG">
                                            <div class="bg-light p-3 my-3">
                                                <ul>
                                                    <li><b>Germline data</b>. Distribution of tumor sites associated
                                                        with the selected mutations;
                                                        number of tumors classified by tumor site divided by total
                                                        number of tumors observed in individuals carriers of the
                                                        selected mutations (% is shown).
                                                    </li>
                                                    <li><b>Somatic data</b>. Distribution of tumor sites associated
                                                        with
                                                        the selected mutations;
                                                        number of mutations classified by tumor site divided by
                                                        total
                                                        number of mutations observed in tumors carrying the selected
                                                        mutations (% is shown).
                                                    </li>
                                                    <li><b>Gene variation data, somatic graph</b>. Proportion of the
                                                        selected mutations among all mutations reported in the
                                                        database
                                                        by tumor sites;
                                                        number of selected mutations classified by tumor site
                                                        divided by
                                                        total number of mutations in the database for each tumor
                                                        sites
                                                        (% is shown).
                                                    </li>
                                                    <li><b>Gene variation data, germline graph</b>. Distribution of
                                                        tumor sites associated with the selected mutations;
                                                        number of tumors classified by tumor site divided by total
                                                        number of tumors observed in individuals carriers of the
                                                        selected mutations (% is shown).
                                                    </li>
                                                    <li><b>Mutation prevalence</b>. Proportion of mutated samples by
                                                        cancer site (topography graph), cancer type (morphology
                                                        graph),
                                                        or by country of origin of the patients (country graph);
                                                        number of mutated samples divided by total number of samples
                                                        analyzed (% is shown).
                                                    </li>
                                                </ul>
                                            </div>
                                        </div>
                                    </li>
                                </ul>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <div id="faq" class="jump-div" style="margin-bottom:40vh;">
            <h2>FAQ</h2>
            <hr>
            <div class="p-3">
                <p>Please note that the <a href="#manual">User's manual</a> provides detailed information on data
                    available and annotation systems.</p>
                <div class="accordion" id="accordionFAQ">
                    <div class="accordion-item">
                        <h2 class="accordion-header" id="headingAnnotations">
                            <button class="accordion-button" type="button" data-bs-toggle="collapse"
                                    data-bs-target="#collapseAnnotations" aria-expanded="true"
                                    aria-controls="collapseAnnotations">
                                Annotations
                            </button>
                        </h2>
                        <div id="collapseAnnotations" class="accordion-collapse collapse show"
                             aria-labelledby="headingAnnotations"
                             data-bs-parent="#accordionFAQ">
                            <div class="accordion-body">
                                <div class="p-3">
                                    <h3><em>TP53</em> reference sequences used in the database</h3>
                                    <hr>
                                    <div>Reference sequences for <em>TP53</em> gene and <em>p53</em> protein can be
                                        found <a
                                                target="_blank" href="resources#sequences">here</a>.
                                    </div>
                                </div>
                                <div class="p-3">
                                    <h3>Where do I find the description of the annotations used in the database?</h3>
                                    <hr>

                                    <div>All annotations are described in details in the <a href="#manual">User's
                                        manual</a>.
                                    </div>
                                </div>
                                <div class="p-3">
                                    <h3>Mutations screened from RNA </h3>
                                    <hr>
                                    <div>All mutations are annotated in the database at the genomic level.
                                        For mutations identified from RNA screening, annotations may not be accurate.
                                        For example, a mutation described as a deletion of exon 5 at RNA level might in
                                        fact be a point mutation located in a splice site at the genomic level
                                        (inducing skiping of an exon). It is of note that this concerns only a small
                                        fraction of the data included in the database. You may exclude studies that
                                        have screened RNA by using the 'Start material' control.
                                    </div>
                                </div>
                                <div class="p-3">
                                    <h3>Description of deletions and insertions</h3>
                                    <hr>
                                    <div> The exact location of deletions, insertions and complex variations are often
                                        poorly described in original reports (often reported at the codon but not
                                        genomic level). Annotations for these mutations are thus not precise since we
                                        annotate mutations at the genomic level. For example, if a deletion is described
                                        as a
                                        deletion of one nucleotide at codon 158, it is entered in the database as
                                        deletion of the
                                        first nucleotide of codon 158 while it may in fact be the second or third
                                        nucleotide that is
                                        actualy deleted. This information is thus only reliable at the codon level.
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="accordion-item">
                        <h2 class="accordion-header" id="headingDA">
                            <button class="accordion-button" type="button" data-bs-toggle="collapse"
                                    data-bs-target="#collapseDA" aria-expanded="true" aria-controls="collapseDA">
                                Data Analysis
                            </button>
                        </h2>
                        <div id="collapseDA" class="accordion-collapse collapse show" aria-labelledby="headingDA"
                             data-bs-parent="#accordionFAQ">
                            <div class="accordion-body">
                                <div class="p-3">
                                    <h3>How can I perform custom analyses? </h3>
                                    <hr>

                                    <div>The analyses that can be performed with the web based tools can not cover all
                                        user needs.
                                        To perform other types of analysis, all datasets are fully
                                        <a href="get_tp53data">downloadable</a> and annotation details are
                                        provided in the <a href="#manual">User's manual</a>.
                                    </div>
                                </div>
                                <div class="p-3">
                                    <h3>How to retrieve a specific <em>p53</em> mutation frequency and residual
                                        activity?</h3>
                                    <hr>
                                    <div>Select the <a href="search_gene_by_mut">Gene variation</a> option;
                                        select a specific mutation with available criteria and click on "Search"
                                    </div>
                                </div>

                                <!--Link broken-->
                                <!--<div class="p-3">-->
                                <!--<h3>Where can I find reports on database analyses?</h3>-->
                                <!--<hr>-->
                                <!--<div>Several analyses of different datasets of the TP53-->
                                <!--database have been published. PubMed links are provided <a-->
                                <!--href="RefsDBanalysis.aspx">here</a>.-->
                                <!--</div>-->
                                <!--</div>-->

                                <div class="p-3">
                                    <h3>How to retrieve the type of somatic mutations found in a specific type of
                                        cancer?</h3>
                                    <hr>
                                    <div>Select the <a href="search_somatic_mut">Somatic mutation</a> option and
                                        select a specific type of tumor with available criteria (ICD-O 3rd
                                        <a href="topo">Topography</a> and <a href="morph">Morphology</a>
                                        classification systems are used for describing tumor types) is used for tumor
                                        classification). The 'Mutation distribution' button leads to a results page with
                                        different types of graphics showing mutation spectra for the selected tumor
                                        type.
                                    </div>
                                </div>
                                <div class="p-3">
                                    <h3>How graphs are calculated?</h3>
                                    <hr>
                                    <div>Several types of graphs are drawn from selected data. See <a class="jump-link"
                                                                                                      href="#collapseMDG">user's
                                        manual</a> for details on graph
                                        calculations.
                                    </div>
                                </div>
                                <div class="p-3">
                                    <h3>How to retrieve the <em>TP53</em> status of a specific cell-line?</h3>
                                    <hr>
                                    <div>Select the <a href="search_cell_lines">Cell lines</a> option; enter the name,
                                        ATCC_ID or characterisitics of a specific cell-line with available criteria;
                                        click on the "Search" button.
                                    </div>
                                </div>
                                <div class="p-3">
                                    <h3>How to retrieve the summary of annotations contained in the database for a
                                        specific mutation?</h3>
                                    <hr>

                                    <div>Select the <a href="search_gene_by_var">Gene variation</a> option; select
                                        a specific mutation with available criteria and click on "Search".
                                        On the result page, click on 'View' button.
                                    </div>
                                </div>
                                <div class="p-3">
                                    <h3>How to retrieve the functional properties and biological activities of specific
                                        mutants?</h3>
                                    <hr>
                                    <div>Select the <a href="search_gene_by_var">Gene variation</a> option; select
                                        a specific mutation with available criteria and click on "Search". On the result
                                        panel, click on 'View' button.
                                    </div>
                                </div>
                                <div class="p-3">
                                    <h3>How to retrieve the type of tumors in which a specific somatic mutation
                                        occurs?</h3>
                                    <hr>
                                    <div>Select the <a href="search_gene_by_var">Gene variation</a> option; select
                                        a specific mutation with available criteria and click on "Search".
                                        On the result page, click on 'Tumor distribution'
                                    </div>
                                </div>
                                <div class="p-3">
                                    <h3>How to retrieve the type of tumors associated with a specific germline
                                        mutation?</h3>
                                    <hr>
                                    <div>Select the <a href="search_germline_mut">Germline mutations</a> option;
                                        select a specific mutation with available criteria and click on "Tumor
                                        distribution".
                                    </div>

                                </div>
                                <div class="p-3">
                                    <h3>How to retrieve the frequency of somatically mutated samples (mutation
                                        prevalence) for (a) specific type(s) of cancer?</h3>
                                    <hr>
                                    <div>Select the <a href="search_somatic_mut">Somatic mutation</a> option;
                                        select a specific type of cancer (and optionally a population and mutation
                                        detection method) and click on the "Search" button.
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="accordion-item">
                        <h2 class="accordion-header" id="headingDBC">
                            <button class="accordion-button" type="button" data-bs-toggle="collapse"
                                    data-bs-target="#collapseDBC" aria-expanded="true" aria-controls="collapseDBC">
                                Database Contents
                            </button>
                        </h2>
                        <div id="collapseDBC" class="accordion-collapse collapse show" aria-labelledby="headingDBC"
                             data-bs-parent="#accordionFAQ">
                            <div class="accordion-body">
                                <div class="p-3">
                                    <h3>Why some mutations present in the somatic dataset are not retrieved with the
                                        'Gene variation' search option?</h3>
                                    <hr>
                                    <div>There may be two reasons for this:</div>
                                    <ol>
                                        <li>Mutations retrieved with this option only include gene variation that are
                                            fully described, while in the dataset of somatic mutations some mutations
                                            are not fully described.
                                        </li>
                                        <li>Somatic mutations may be reported in individuals with different SNP status.
                                            If a mutation is close to a SNP, it may have a different impact on the
                                            protein sequence depending on the SNP status. For example, the mutation
                                            c.637C&gt;T on the first base of codon 213, will
                                            result in a p.R213X change in the protein
                                            sequence if the SNP present on the third base of the codon is a A (CGA&gt;TGA),
                                            while it will result in a
                                            p.R213W change if the SNP present on the third base of the codon is a G (CGG&gt;TGG).
                                            Mutations not described from the reference
                                            sequence are included in the somatic dataset, but not in the Gene variation
                                            dataset.
                                        </li>
                                    </ol>
                                </div>
                                <div class="p-3">
                                    <h3>Why mutation numbers retrieved from the mutation prevalence and mutation
                                        spectrum datasets are different?</h3>
                                    <hr>

                                    <p> Data in the prevalence dataset are "independent" from data included
                                        in the somatic dataset. Numbers differ for the following reasons:
                                    </p>

                                    <ul>
                                        <li>Because we retrieve data from papers, and in many papers the only
                                            information that can be
                                            extracted is the total number of samples analyzed and total number of
                                            samples mutated (mutations
                                            are not described in details), mutations can not be included in the somatic
                                            dataset. Thus, numbers
                                            in the prevalence table do not match numbers in the somatic dataset
                                            (mutation spectrum).
                                        </li>

                                        <li>Numbers by histologies may also differ, as for example, a paper may contain
                                            mutation details for lung
                                            ADC, SCC, LCC (which are all non-small cell lung cancers), but total numbers
                                            of samples analyzed for each
                                            histology is not available. In this case, mutations corresponding to ADC,
                                            SCC and LCC will be entered in
                                            the somatic dataset but in the prevalence table the prevalence will be
                                            indicated only for non-small cell
                                            carcinoma (group that includes the 3 tumor types).
                                        </li>

                                        <li> Cell-lines are not included in the prevalence count.</li>

                                        <li> Samples with more than one mutations are counted once in the prevalence
                                            table
                                            while all mutations are entered in the somatic dataset.
                                        </li>

                                        <li> The prevalence may be missing for some papers that describe mutations
                                            included in
                                            the somatic dataset. The prevalence dataset has been added in a recent
                                            version of the database
                                            (2001 while the database started in 1994) and not all papers have been
                                            reviewed. The non-reviewed
                                            papers correspond mainly to publications that describe less than 10
                                            mutations (about 400 papers).
                                            For some papers, the prevalence could not be retrieved from the information
                                            provided in the
                                            publication (about 100 papers).
                                        </li>
                                    </ul>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </main>

</div>

{% endblock %}
