{% extends "base.html" %}
{% block title %}Other Resources{% endblock %}
{% block head %}
{{ super() }}
<script type="text/javascript">
    active_menu = 'resources';
</script>

<script type="text/javascript" src="{{ url_for('static', filename='js/libs/molstar.js') }}"></script>

{% endblock %}

{% block content %}
<main class="bg-white">
    <div class="page-banner">
        <div class="container">
            <h1 class="fw-bold p-2 my-0">Other Resources</h1>
        </div>
    </div>
    <div class="container">
        <nav class="sidebar mt-3">
            <div class="py-2">
                <div class="px-3 fw-bold">On this page:</div>
                <ul class="nav flex-column">
                    <li class="nav-item">
                        <a href="#sequences"><em>TP53</em> Sequences</a>
                    </li>
                    <li class="nav-item">
                        <a href="#mutation">Variant Nomenclatures</a>
                    </li>
                    <li class="nav-item">
                        <a href="#tools">Protocols and Tools</a>
                    </li>
                    <li class="nav-item">
                        <a href="#analysis">3D Structures Analysis</a>
                    </li>
                    <li class="nav-item">
                        <a href="#functions">Definitions and Functions</a>
                    </li>
                    <li class="nav-item">
                        <a href="#others">Other Databases</a>
                    </li>
                    <li class="nav-item">
                        <a href="#books">Books</a>
                    </li>

                </ul>
            </div>
        </nav>
        <div class="p-2">
            <div id="sequences" class="mt-3">
                <h2><em>TP53</em> Sequences</h2>
                <hr>
                <div class="p-3">
                    <h3>Reference sequences used to annotate <em>TP53</em> gene variations</h3>
                    <ul>
                        <li>Graphical view of the <a href="/hg38seq">Genomic sequence (hg38)</a>
                            with exon-intron boundaries for the main <em>p53</em> isoform.
                        </li>

                        <li><a href="/tp53seq">Coding and protein sequence</a> with residue numbers,
                            amino-acids, highlighted CpG sites and exon-intron boundaries.
                        </li>

                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.ncbi.nlm.nih.gov/nuccore/NM_000546">
                                NCBI link to cDNA sequence</a> corresponding to the NM_000546 isoform.
                        </li>

                    </ul>
                </div>
                <div class="p-3">
                    <h3>Other links to human sequences</h3>
                    <ul>
                        <li><a href="/p53isoforms"><em>p53</em> isoforms</a>
                        </li>

                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.ncbi.nlm.nih.gov/sites/entrez?db=gene&cmd=search&term=7157">
                                NCBI EntrezGene</a>
                        </li>

                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.ensembl.org/Homo_sapiens/geneview?db=&amp;gene=ENSG00000141510">Gene
                                view at
                                ENSEMBL</a>
                        </li>

                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.uniprot.org/uniprot/P04637">Protein annotations at
                                UniProtKB</a>
                        </li>

                    </ul>
                </div>
                <div class="p-3">
                    <h3>Sequences in other species</h3>
                    <ul>
                        <li><a href="/seq_align">Protein sequence alignment</a></li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.rcsb.org/pdb/explore/explore.do?structureId=1T4W">Structure
                                of
                                <em>p53</em> C.
                                Elegans Ortholog Cep-1</a>
                        </li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.rcsb.org/pdb/explore/explore.do?structureId=1HU8">Structure
                                of
                                Mouse <em>p53</em> core domain</a>
                        </li>
                    </ul>
                </div>
                <div class="p-3">
                    <h3>3D Sequence Structures</h3>
                    <ul>
                        <li><a href="{{ url_for('static', filename='download/P53DNA.zip') }}">Download <em>p53</em>
                                core
                                domain and
                                Rasmol viewer</a></li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.rcsb.org/pdb/explore/explore.do?structureId=1TSR">
                                <em>p53</em> core domains bound to DNA</a>
                        </li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.rcsb.org/pdb/explore/explore.do?structureId=1YCS">
                                <em>p53</em> core domain bound to the ankyrin and SH3 domains of 53BP2</a>
                        </li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.rcsb.org/pdb/explore/explore.do?structureId=1KZY">
                                <em>p53</em> core domain bound to the Brct Region of 53Bp1</a>
                        </li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.rcsb.org/pdb/explore/explore.do?structureId=1GZH">
                                <em>p53</em> core domains (2) bound to the Brct Regions (2) of 53Bp1</a>
                        </li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.rcsb.org/pdb/explore/explore.do?structureId=1AIE">
                                <em>p53</em> tetramerization domain (residues 326-356)</a>
                        </li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.rcsb.org/pdb/explore/explore.do?structureId=1C26">
                                <em>p53</em> tetramerization domain (X-ray)</a>
                        </li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.rcsb.org/pdb/explore/explore.do?structureId=1YCR">
                                <em>p53</em> transactivation domain bound to mdm2</a>
                        </li>
                    </ul>
                </div>
            </div>
            <div id="mutation" class="mt-5">
                <h2>Variant Nomenclatures</h2>
                <hr>
                <div class="p-3">
                    <div>The genomic reference sequence used for annotating <em>TP53</em> variants is <a
                            href="/hg38seq">NC_000017.11 (hg38 build)</a> from GenBank.
                    </div>
                    <div>
                        Variants are reported according to <a
                            href="https://varnomen.hgvs.org/recommendations/DNA/variant/substitution/" target="_blank"
                            rel="noopener noreferrer">HGVS</a> nomenclature recommendations.
                    </div>
                    <div>
                        We recommend to use <a target="_blank" rel="noopener noreferrer"
                            href="https://www.mutalyzer.nl/">Mutalyzer</a>, to help
                        you
                        format your variant data according to HGVS guidelines and to convert positions between
                        different
                        genome build or isoforms.
                    </div>
                </div>
            </div>
            <div id="tools" class="mt-5">
                <h2>Protocols and Tools</h2>
                <hr>
                <div class="p-3">
                    <h3>Protocols</h3>
                    <div class="pt-2">
                        To assess <em>TP53</em> variant status it is recommended to perform gene sequencing of all
                        exons,
                        including
                        the splice junctions and completed with MLPA to detect large deletion/duplications.
                        Immuno-histochemical analysis is not a good method for assessing <em>TP53</em> variant
                        status,
                        as
                        it cannot
                        distinguish between the various types of variants, does not detect truncating variants
                        (false-negative results), and gives
                        false-positive results (over-expressed WT protein).
                    </div>
                    <div class="pt-3">
                        Some Example of Sequencing Protocols:
                        <ul>
                            <li>Direct Sequencing by Sanger: <a id="sanger_seq_protocol_link"
                                    href="/pdf/TP53_SangerSequencing_NIH" class="ps-2" target="_blank"
                                    rel="noopener noreferrer"><i class="far fa-file-pdf"></i>
                                    TP53_SangerSequencing_NIH.pdf</a>
                            </li>
                            <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/27377626" target="_blank"
                                    rel="noopener noreferrer"><em>TP53</em>
                                    sequencing by a next generation sequencing technology (NGS)</a>. The detailed list
                                of primers can be
                                downloaded here:
                                <a class="ps-2" href="{{ url_for('static', filename='download/NGS_TP53_Primers.txt') }}"
                                    target="_blank" rel="noopener noreferrer">
                                    <i class="far fa-file-alt"></i> NGS_TP53_Primers.txt
                                </a>
                            </li>
                            <li><a href="https://www.mrcholland.com/technology/mlpa/technique" target="_blank"
                                    rel="noopener noreferrer">MLPA</a>
                            </li>
                            <li>Some caution regarding <em>TP53</em> variants detected by multigene panel testing by
                                NGS: a
                                study by
                                <a target="_blank" rel="noopener noreferrer"
                                    href="https://www.ncbi.nlm.nih.gov/pubmed/?term=29189820">
                                    Weitzel et al.</a>
                            </li>
                            <li>Sequencing and Haplotyping: <a href="/pdf/TP53_SNP_sequencing" class="ps-2"
                                    target="_blank" rel="noopener noreferrer"><i class="far fa-file-pdf"></i>
                                    TP53_SNP_sequencing.pdf</a>
                            </li>
                        </ul>
                    </div>
                    <div class="pt-3">
                        Recent developments on <em>p53</em> antibodies
                        <ul>

                            <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/29298430" target="_blank"
                                    rel="noopener noreferrer">Monoclonal
                                    Antibodies
                                    against
                                    Specific <em>p53</em> Hotspot Mutants as Potential Tools for Precision
                                    Medicine</a><br />
                            </li>
                            <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/30907951" target="_blank"
                                    rel="noopener noreferrer">Understanding
                                    <em>p53</em>
                                    functions
                                    through <em>p53</em> antibodies</a>
                            </li>

                        </ul>
                    </div>
                </div>
                <div class="p-3">
                    <h3>Other tools</h3>
                    <div class="pt-2">
                        <ul>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://www.mutalyzer.nl/">Mutalyzer</a>,
                                a package to support
                                checks
                                of
                                sequence variant nomenclature according to the guidelines of the Human Genome
                                Variation
                                Society
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://www.niehs.nih.gov/research/resources/databases/p53/">Resources
                                    for
                                    the systematic analysis of <em>p53</em> interactions with DNA and other
                                    transcription
                                    factors</a>.
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://tomateba.github.io/p53retriever/">p53retriever</a>, for
                                the
                                detection and display of <em>p53</em> putative response elements (REs) on DNA
                                sequences
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="http://www.targetgenereg.org/">TargetGeneReg</a>,
                                for
                                displaying
                                gene
                                of
                                interest in the <em>p53</em> and cell cycle gene regulatory networks
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="http://gepia.cancer-pku.cn/detail.php?gene=TP53"><em>TP53</em>
                                    RNA
                                    expression data from TCGA at GEPIA</a></li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://datamed.org/search?datamed_dataset_index_entities%5Bquery%5D=TP53">Datamed</a>,
                                a
                                search engine for <em>TP53</em> datasets
                            </li>
                            <!--<li><a target="_blank" rel="noopener noreferrer" href="https://www.genomnis.com/access-hsf">Human Splicing Finder</a></li>-->
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="http://agvgd.hci.utah.edu/">Align-AGVGD</a>,
                                for the prediction
                                of
                                the
                                functional impact of missense variants
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://sift.bii.a-star.edu.sg/">SIFT</a>,
                                for the prediction
                                of
                                the
                                functional impact of missense variants
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://www.ebi.ac.uk/thornton-srv/databases/cgi-bin/valdar/scorecons_server.pl">Scorecons</a>,
                                for scoring residue conservation
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://www.cancergenomeinterpreter.org/home">The Cancer Genome
                                    Interpreter</a>, to identify driver variants in tumor samples
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://bioinformatics.mdanderson.org/public-software/lfspro/">LFSPRO</a>,
                                a tool for estimating <em>TP53</em> Variant Carrier Probability
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://bioinformatics.mdanderson.org/public-software/famdenovo/">Famdenovo</a>,
                                an
                                algorithm that calculates the probability of a germline variant to be de novo based
                                on
                                family
                                history data
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer" href="/aa_codes">Amino-acid letter codes
                                    and properties</a></li>
                            <li><a target="_blank" rel="noopener noreferrer" href="/genetic_code">Genetic code</a>
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer" href="https://www.lovd.nl/2.0/">LOVD, a
                                    software to build variant
                                    databases</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
            <div id="analysis" class="mt-5">
                <h2>3D Structures Analysis</h2>
                <hr>
                <div class="p-3">
                    <h3 class="pt-3">3D Viewer</h3>
                    <div>
                        <div class="text-white small-font bg-secondary px-2 three_d_viwer"><i
                                class="fas fa-info-circle"></i> Use mouse or control buttons (located on the right)
                            to
                            explore.
                        </div>
                        <div id="molstarViewer" class="three_d_viwer"></div>
                        <script type="text/javascript">
                            var viewer = new molstar.Viewer('molstarViewer', {
                                layoutIsExpanded: false,
                                layoutShowControls: false,
                                layoutShowRemoteState: false,
                                layoutShowSequence: true,
                                layoutShowLog: false,
                                layoutShowLeftPanel: true,
                                viewportShowExpand: true,
                                viewportShowSelectionMode: false,
                                viewportShowAnimation: false,
                                pdbProvider: 'rcsb',
                            });
                            viewer.loadStructureFromUrl("{{ url_for('static', filename='pdb/1tsr_B.pdb') }}", format = 'pdb');
                        </script>
                    </div>
                    <h3 class="pt-3">Structural Impacts of Variants</h3>
                    <div>
                        <ul>
                            <li> Analysis by A.C. Martin: <a target="_blank" rel="noopener noreferrer"
                                    href="https://www.ncbi.nlm.nih.gov/pubmed/11793474">Publication</a>
                            </li>

                            <li> Analysis by <a target="_blank" rel="noopener noreferrer"
                                    href="https://www.ncbi.nlm.nih.gov/pubmed/16395672">Mathe
                                    (2006)</a>.
                                See details here:
                                <a target="_blank" rel="noopener noreferrer" href="/dt_method">Structure-Function
                                    Predictions Based on Scores Derived from Delaunay Tessellations</a>
                            </li>

                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://www.uniprot.org/uniprot/P04637#section_features">UniProt
                                    information</a>
                            </li>

                            <li>Stability and oligomeric structure of <em>p53</em> tetramerization domain mutants by
                                <a target="_blank" rel="noopener noreferrer"
                                    href="https://www.ncbi.nlm.nih.gov/pubmed/20978130">Kamada
                                    et al. (2011)</a>
                            </li>

                            <li>Solved crystal structures:</li>
                            <dt>by <a target="_blank" rel="noopener noreferrer"
                                    href="https://www.ncbi.nlm.nih.gov/pubmed/15703170">Joerger
                                    AC et al. (2005)</a>
                            </dt>

                            <ul>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=1UOL">M133L/V203A/N239Y/N268D</a>
                                </li>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2BIM">M133L/V203A/N239Y/N268D/R273H</a>
                                </li>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2BIN">M133L/H168R/V203A/N239Y/N268D</a>
                                </li>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2BIO">M133L/V203A/N239Y/R249S/N268D</a>
                                </li>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2BIP">M133L/H168R/V203A/N239Y/R249S/N268D</a>
                                </li>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2BIQ">T123A/M133L/H168R/V203A/N239Y/R249S/N268D</a>
                                </li>
                            </ul>
                            <dt>by <a target="_blank" rel="noopener noreferrer"
                                    href="https://www.ncbi.nlm.nih.gov/pubmed/17015838">Joerger AC et
                                    al.
                                    (2006)</a></dt>
                            <ul>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2J1W">M133L/V143A/V203A/N239Y/N268D</a>
                                </li>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2J1X">M133L/V203A/Y220C/N239Y/N268D</a>
                                </li>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2J1Y">M133L/V203A/G245S/N239Y/N268D</a>
                                </li>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2J1Z">M133L/V203A/N239Y/N268D/F270L</a>
                                </li>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2J20">M133L/V203A/N239Y/N268D/R273C</a>
                                </li>
                                <li><a target="_blank" rel="noopener noreferrer"
                                        href="https://www.rcsb.org/pdb/explore/explore.do?structureId=2J21">M133L/V203A/N239Y/N268D/R282W</a>
                                </li>
                            </ul>
                            <li><a href="/stability">Thermodynamic Stability Data</a>
                        </ul>
                    </div>
                </div>
            </div>
            <div id="functions" class="mt-5">
                <h2>Definitions and Functions</h2>
                <hr>
                <div class="p-3">
                    <ul>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.ncbi.nlm.nih.gov/gene/7157"><em>TP53</em> entry at
                                EntrezGene</a>
                        </li>
                        <li><a target="_blank" rel="noopener noreferrer" href="https://omim.org/entry/191170">OMIM
                                review at NCBI</a></li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=TP53&search=TP53"><em>TP53</em>
                                GeneCards entry</a></li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://pubchem.ncbi.nlm.nih.gov/pathway/PathBank:SMP0063782">P53 Signaling Pathway</a></li>
                        <li><a href="target_genes"><em>p53</em> target genes and response-elements</a></li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.ensembl.org/Homo_sapiens/geneview?db=&amp;gene=ENSG00000141510">Gene
                                annotation by Ensembl at EBI</a></li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://atlasgeneticsoncology.org/Genes/P53ID88.html"><em>p53</em>
                                entry at
                                Atlas of
                                Genetics</a></li>
                        <!-- <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.hprd.org/interactions?hprd_id=01859&isoform_id=01859_1&isoform_name="><em>p53</em>
                                protein interactions at Human Protein Reference Database</a></li> -->
                        <!-- <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.hprd.org/ptms?hprd_id=01859&isoform_id=01859_1&isoform_name=Isoform_1"><em>p53</em>
                                post-translational modifications at Human Protein Reference Database</a></li> -->
                        <li><a href="/p53isoforms"><em>p53</em> isoforms</a></li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://www.uniprot.org/uniprot/P04637">Protein annotation by
                                SwissProt/UniProtKB</a></li>
                        <li><a target="_blank" rel="noopener noreferrer"
                                href="https://p53.fr/tp53-information/historical-aspects"><em>p53</em>
                                history and
                                info by
                                Thierry Soussi</a></li>
                        <li><em>TP53</em> family memmbers: <a target="_blank" rel="noopener noreferrer"
                                href="https://www.ncbi.nlm.nih.gov/gene/8626"><em>TP63</em></a>
                            and
                            <a target="_blank" rel="noopener noreferrer"
                                href="https://www.ncbi.nlm.nih.gov/gene/7161"><em>TP73</em></a>
                        </li>
                    </ul>
                    <h3><em>p53</em> in the Clinic</h3>
                    <div class="pt-2">
                        <ul>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://www.ncbi.nlm.nih.gov/books/NBK1311/">Li-Fraumeni
                                    syndrome
                                    at
                                    GeneReviews</a></li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://videocast.nih.gov/summary.asp?Live=9741">Videocast of
                                    the
                                    Li-Fraumeni Syndrome Workshop 2010</a></li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://www.clinicaltrials.gov/ct2/results?cond=&term=p53"><em>p53</em>
                                    in
                                    clinical
                                    trials at ClinicalTrials.gov</a></li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://www.springer.com/biomed/cancer/book/978-1-4614-3675-1">"<em>p53</em>
                                    in
                                    the
                                    clinics" book, 2012</a></li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://bioinformatics.mdanderson.org/main/LFSPRO">LFSPRO</a>,
                                a
                                tool
                                for
                                estimating <em>TP53</em> Variant Carrier Probability
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://bioinformatics.mdanderson.org/public-software/famdenovo/">Famdenovo</a>,
                                an
                                algorithm that calculates the probability of a germline variant to be de novo,
                                taking
                                <em>TP53</em>
                                as
                                an
                                example application
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
            <div id="others" class="mt-5">
                <h2>Other Databases</h2>
                <hr>
                <div class="p-3">
                    <h3><em>TP53</em> databases</h3>
                    <div class="pt-2">
                        <p>There are several links to different <em>TP53</em> databases available on the web.
                            Several
                            resources are
                            not updated anymore and may contain obsolete data. It is recommended to use up-to-date
                            resources.
                        </p>
                        <ul>
                            <li><a target="_blank" rel="noopener noreferrer" href="/">The <em>TP53</em> Database</a>
                                (last updated in July 2019)
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer" href="https://p53.fr/">The <em>p53</em>
                                    database by Thierry
                                    Soussi</a> (last
                                updated in 2017)
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://databases.lovd.nl/shared/genes/TP53">LOVD <em>TP53</em></a>,
                                a
                                database system for direct data submission
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://www.lf2.cuni.cz/projects/germline_mut_p53.htm">The
                                    database
                                    of germline <em>TP53</em> mutations in Prague</a> (not updated anymore)
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="http://p53.bii.a-star.edu.sg/index.php"><em>p53</em>
                                    Knowledgebase at
                                    Singapore</a> (obsolete)
                            </li>
                            <li><a target="_blank" rel="noopener noreferrer"
                                    href="https://www.ibiblio.org/dnam/des_p53.htm">Human <em>p53</em>
                                    database and
                                    software, N. Cariello (1997)</a> (obsolete)
                            </li>
                        </ul>
                    </div>
                    <h3>Related Databases</h3>
                    <div class="pt-2">
                        <ul>
                            <li><a href="https://cancer.sanger.ac.uk/cosmic" target="_blank"
                                    rel="noopener noreferrer">Catalogue Of Somatic
                                    Mutations
                                    in
                                    Cancer (COSMIC) at Sanger Institute, UK</a></li>
                            <li><a href="https://dcc.icgc.org/" target="_blank" rel="noopener noreferrer">International
                                    Cancer Genome Consortium
                                    (ICGC)
                                    data portal</a></li>
                            <li><a href="https://www.cbioportal.org/" target="_blank"
                                    rel="noopener noreferrer">cBioPortal
                                    for Cancer Genomics</a>
                            </li>
                            <li><a href="https://genie.cbioportal.org/" target="_blank" rel="noopener noreferrer">AACR
                                    GENIE dataset in
                                    cBioportal</a>
                            </li>
                            <li><a href="https://portal.gdc.cancer.gov/" target="_blank" rel="noopener noreferrer">TCGA
                                    data portal</a></li>
                            <li><a href="https://grenada.lumc.nl/LSDB_list/lsdbs" target="_blank"
                                    rel="noopener noreferrer">List of locus specific
                                    databases</a></li>
                            <li><a href="https://pecan.stjude.cloud/" target="_blank" rel="noopener noreferrer">St.
                                    Jude PeCan Data Portal on
                                    pediatric cancer mutations
                                </a></li>
                        </ul>
                    </div>
                </div>
            </div>
            <div id="books" class="mt-5">
                <h2>Books</h2>
                <hr>
                <div class="p-3">
                    <div class="pt-2 row">
                        <div class="col-auto d-none d-lg-block">
                            <a href="/tp53book2013"><img height="150"
                                    src="{{ url_for('static', filename='img/p53_2013.jpeg') }}"
                                    alt="p53 in the Clinics: Book Cover"></a>
                        </div>
                        <div class="col px-4">
                            <h3><em>p53</em> in the Clinics</h3>
                            With over 60,000 referenced publications, <em>p53</em> has emerged as one of the
                            most important factors in human cancer. Research on <em>p53</em> has led to a complete
                            overhaul of our understanding of the molecular basis of human cancer.
                        </div>

                    </div>
                </div>
                <div class="p-3">

                    <div class="pt-2 row">
                        <div class="col-auto d-none d-lg-block">
                            <a href="/tp53book"><img height="150" alt="25 Years of p53 Research: Book Cover"
                                    src="{{ url_for('static', filename='img/25yrs_p53.webp') }}">
                            </a>
                        </div>
                        <div class="col px-4 ">
                            <h3>25 Years of <em>p53</em> Research</h3>
                            The discovery of <em>p53</em> in 1979 marks the beginning of a most fascinating era of
                            modern
                            cancer research and molecular biology, an era that is still in full swing and does not
                            show
                            any
                            signs of ending in the foreseeable future.
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
</main>

{% endblock %}