{% extends "tp53data.html" %}
{% block title %}TP53 Database: Search Results on Gene Variants{% endblock %}
{% block head %}
{{ super() }}
<script type="text/javascript">
    let show_hg19 = (assign_var("{{ criteria }}").indexOf("&#39;g_description&#39;") > 0|false);
</script>

<script type="text/javascript" src="{{ url_for('static', filename='js/libs/select.dataTables.min.js') }}"></script>
<script type="text/javascript" src="{{ url_for('static', filename='js/libs/dataTables.buttons.min.js') }}"></script>
<script type="text/javascript" src="{{ url_for('static', filename='js/libs/buttons.html5.min.js') }}"></script>
<script type="text/javascript" src="{{ url_for('static', filename='js/libs/dataTables.fixedColumns.min.js') }}"></script>
<script type="text/javascript" src="{{ url_for('static', filename='js/results_gene_mutation.js') }}"></script>

{% endblock %}

{% block subcontent %}
<div class="page-banner">
    <div class="container">
        <h1 class="fw-bold p-2 my-0">Search Results: <span class="ms-2 fw-light">Functional / Structural Data</span></h1>
    </div>
</div>
<div class="m-5">
    <div id="gm-result" class="my-4">
        <div class="my-2">
            <a class="text-start btn btn-sm btn-outline-dark criteria-btn collapsed"
               data-bs-toggle="collapse" href="#criteria_div" role="button" aria-label="View/Hide Criteria"
               aria-expanded="false"
               aria-controls="criteria_div"></a>
            <button type="button" title="Download Data in .csv"
                    class="btn btn-sm darkblue-btn download-btn ms-2">
                <i class="fas fa-download"></i>
                Download
            </button>
            <button type="button" class="btn btn-sm darkblue-btn action-button"
                    value="get_mutation_dist" title="View Variant Distribution"><i
                    class="fas fa-chart-pie"></i>
                Variant Distribution
            </button>
            <button type="button" class="btn btn-sm darkblue-btn action-button"
                    value="get_gv_tumor_dist" title="View Tumor Distribution"><i
                    class="fas fa-chart-bar"></i>
                Tumor Distribution
            </button>
            <button type="button" class="btn btn-sm darkblue-btn cell-search-button disabled me-1"
                    title="View Cell Lines"><i
                    class="fas fa-search"></i>
                Cell Lines <span
                        class="cell-line-count-badge badge rounded-pill bg-secondary bg-light darkblue-text"
                        style="font-size: .75rem;">0</span>
            </button>
            <div class="btn-group btn-group-sm ms-1" title="No of selected rows">
                <div class="pb-1"><em>Selected Rows</em> <span
                        class="badge rounded-pill bg-secondary cart-count">0</span></div>
            </div>
            {% include 'criteria_div.html' %}
        </div>
        <div>
            <table id="gm-result-table" class="table table-striped table-sm result-table small-font" cellspacing="0"
                   width="100%">
                <thead>
                <td data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Select all rows on this page">
                    <input class="form-check-input check-all" type="checkbox"
                           aria-label="Select All">
                </td>
<!--                <th data-bs-toggle="tooltip"-->
<!--                    data-bs-placement="top"-->
<!--                    title="View variant details">-->
<!--                    Variant<br/>Details-->
<!--                </th>-->

<!--                <th data-bs-toggle="tooltip"-->
<!--                    data-bs-placement="top"-->
<!--                    title="View variant details">-->
<!--                    View Variant Details-->
<!--                </th>-->
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Mutation nomenclature according to HGVS standards and using the GenBank NC_000017.10 (hg19 assembly) genomic sequence as reference">
                    Genomic Description (hg19)
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Mutation nomenclature according to HGVS
                    standards and using the GenBank NC_000017.11 (hg38 assembly) genomic sequence as reference.">
                    Genomic Description (hg38)
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Mutation nomenclature according to HGVS standards and using the NM_000546.5 coding sequence as reference.">
                    cDNA Description
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Mutation description at the protein level as recommended by HGVS and using the Uniprot reference sequence P04637.">
                    Protein Description
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Location of the mutation in the introns or exons in TP53 gene for the reference
                    sequence NM_000546.5. Terms occurring in this column are '1-intron' to '11-intron' and
                    '2-exon' to '11-exon'. An 'i' or 'e' in front mean that the mutation is located
                    within the indicated intron or exon with no information on the precise location.">
                    Exon No
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Effect of the mutation. The terms occurring in this column are: missense (change of
                    one amino-acid), nonsense (introduction of a stop codon), FS (frameshift), silent (no
                    change in the protein sequence), splice (mutations located in the two first and two
                    last conserved nucleotides of the introns and are thus predicted to alter splicing,
                    or mutations that have been shown to alter splicing experimentally), other (inframe
                    deletions or insertions), intronic (mutations in introns outside splicing sites), NA
                    (mutations upstream in 5' or 3' UTR).">
                    Effect
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Functional classification based on the overall transcriptional activity (TA) on 8
                    different promoters as measured in yeast assays by Kato et al. For each mutant, the median of the 8 promoter-specific activities
                    (expressed as percent of the wild-type protein) is calculated and missense mutations are classified
                    as 'non-functional' if the median is less or equal to 20, 'partially functional' if the median is greater than 20 and
                    less than or equal to 75, 'functional' if the median is greater than 75 and less than or equal to 140, and 'supertrans' if the median is
                    greater than 140.">
                    TA Class
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Functional classification for loss of growth-suppression and
                                                                    dominant-negative activities based on Z-scores from
                                                                    Giacomelli et al., (2018) study:
                                                                    DNE_LOF = p53WTNutlin3 Z-score >= 0.61 and
                                                                    Etoposide
                                                                    Z-score
                                                                    <=
                                                                    -0.21;
                                                                    notDNE_notLOF = p53WTNutlin3 Z-score &lt; 0.61
                                                                    and
                                                                    Etoposide
                                                                    Z-score
                                                                    &gt; -0.21;
                                                                    notDNE_LOF if p53WTNutlin3 Z-score &lt; 0.61 and
                                                                    Etoposide
                                                                    Z-score
                                                                    &lt;=
                                                                    -0.21; unclass = others">
                    DNE / LOF Class
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Missense variant functional predictions by an optimized Align-GVGD tool. Mutations
                    classified as 'C0' are considered tolerated while other classes are considered damaging.
                    Further details in 
                    Fortuno (2018).">
                    Align-GVGD Class
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Number of occurrence in the somatic dataset (number of tumors reported to carry
                    this tumor variant). Total count is 29,891 in R20.">
                    Somatic Count
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Number of occurrence in the germline dataset (number of pedigree/individual carriers of this germline variant). Total count is 1,532 in R20.">
                    Germline Count
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Number of occurrence in the cell-line dataset (number of cell-lines reported to carry this mutation). Total cell-line count is 2,766 in R20.">
                    CellLine Count
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Sum of variation occurrence from TCGA (MC3), ICGC (v28) and GENIE (V5) datasets. Total count is 23,570.">
                    TCGA ICGC GENIE Count
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Number of occurrence in GDC available.">
                    GDC Case Count
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Polymorphic status of the gene variation.'Validated': MAF < 0.001 in ESP6500, 1000G or gnomAD databases;
                    'No': not reported or reported at MAF < 0.001 in ESP6500, 1000G, or gnomAD databases;
                    'NA': not applicable.">
                    Validated SNP
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Link to ClinVar database.">
                    ClinVar
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Link to mutation ID in COSMIC database.">
                    COSMIC
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Link to NCBI SNP database.">
                    dbSNP
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Link to gnomAD database.">
                    gnomAD
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    data-bs-html="true"
                    title="The highest delta score of a variant for <strong>acceptor gain (AG)</strong>, ranging from 0 to 1 and can be interpreted as the probability that the variant affects splicing at any position within a window around it (10kb as maximum distance, as recommended for best performance <span class='fst-italic'> [references: PMID 34283047, PMID 33942434]</span>). The maximum of four delta scores (DS_AG, DS_AL, DS_DG, DS_DL) is used as basis for spliceogenicity predictions.">
                    SpliceAI DS_AG
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    data-bs-html="true"
                    title="The highest delta score of a variant for <strong>acceptor loss (AL)</strong>, ranging from 0 to 1 and can be interpreted as the probability that the variant affects splicing at any position within a window around it (10kb as maximum distance, as recommended for best performance <span class='fst-italic'> [references: PMID 34283047, PMID 33942434]</span>). The maximum of four delta scores (DS_AG, DS_AL, DS_DG, DS_DL) is used as basis for spliceogenicity predictions.">
                    SpliceAI DS_AL
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    data-bs-html="true"
                    title="The highest delta score of a variant for <strong>donor gain (DG)</strong>, ranging from 0 to 1 and can be interpreted as the probability that the variant affects splicing at any position within a window around it (10kb as maximum distance, as recommended for best performance <span class='fst-italic'> [references: PMID 34283047, PMID 33942434]</span>). The maximum of four delta scores (DS_AG, DS_AL, DS_DG, DS_DL) is used as basis for spliceogenicity predictions.">
                    SpliceAI DS_DG
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    data-bs-html="true"
                    title="The highest delta score of a variant for <strong>donor Loss (DL)</strong>, ranging from 0 to 1 and can be interpreted as the probability that the variant affects splicing at any position within a window around it (10kb as maximum distance, as recommended for best performance <span class='fst-italic'> [references: PMID 34283047, PMID 33942434]</span>). The maximum of four delta scores (DS_AG, DS_AL, DS_DG, DS_DL) is used as basis for spliceogenicity predictions.">
                    SpliceAI DS_DL
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    data-bs-html="true"
                    title="The location of <strong>acceptor gain (AG)</strong> relative to variant position in the pre-mRNA transcript (positive values are upstream of the variant, negative values are downstream).">
                    SpliceAI DP_AG
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    data-bs-html="true"
                    title="The location of <strong>acceptor loss (AL)</strong> relative to variant position in the pre-mRNA transcript (positive values are upstream of the variant, negative values are downstream).">
                    SpliceAI DP_AL
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    data-bs-html="true"
                    title="The location of <strong>donor gain (DG)</strong> relative to variant position in the pre-mRNA transcript (positive values are upstream of the variant, negative values are downstream).">
                    SpliceAI DP_DG
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    data-bs-html="true"
                    title="The location of <strong>donor loss (DL)</strong> relative to variant position in the pre-mRNA transcript (positive values are upstream of the variant, negative values are downstream).">
                    SpliceAI DP_DL
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="View variant details">
                    Variant<br/>Details
                </th>
                </tr>
                </thead>
            </table>
        </div>
    </div>
</div>
{% endblock %}
