{% extends "tp53data.html" %}
{% block head %}
{{ super() }}
<script type="text/javascript">
    active_submenu = 'view_val_poly';
</script>
<script type="text/javascript" src="{{ url_for('static', filename='js/libs/dataTables.buttons.min.js') }}"></script>
<script type="text/javascript" src="{{ url_for('static', filename='js/libs/buttons.html5.min.js') }}"></script>
<script type="text/javascript" src="{{ url_for('static', filename='js/view_val_poly.js') }}"></script>


{% endblock %}

{% block subcontent %}
<div class="pt-4">
    <h2>View Validated Polymorphisms</h2>
    <hr>

    <div id="view_val_poly" class="p-3">

        <div class="text-muted pb-2">
            Polymorphisms listed here are variants that have been observed in unaffected human
                population at frequency > 0.001 in at least one of the following databases: 1000G, gnomAD, or ESP6500.<br>
            Go to <a href="refs_corner#polymorph"> Refs Corner > <em>TP53</em>
                Polymorphisms</a> to learn more about the functional impact and the disease associations.
        </div>

        <div class="py-2">
            <!--<a type="button" title="Read more about TP53 Polymorphisms: the functional impact and the disease associations" class="float-end btn btn-sm btn-outline-secondary"-->
               <!--href="refs_corner#polymorph"><i class="fas fa-arrow-circle-right"></i> Read More: TP53-->
                <!--Polymorphisms</a>-->
            {% include 'criteria_div.html' %}
        </div>


        {% if query_result.data|length %}
        <div class="float-end">
            <div class="btn-group btn-group-sm">
                <button type="button" class="btn btn-outline-secondary download-btn"><i class="fas fa-download"></i>
                    Download
                </button>
            </div>
        </div>
        <table id="val-poly-result-table" class="table table-striped table-sm small-font result-table clientside-processed"
               cellspacing="0" width="100%">
            <thead>
            <tr>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Mutation nomenclature according to HGVS standards and using the GenBank NC_000017.11 (hg38 assembly) genomic sequence as reference.">
                    Genomic Description (hg38)
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Mutation nomenclature according to HGVS standards and using the NM_000546.5 coding sequence as reference.">
                    cDNA Description
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Mutation description at the protein level as recommended by HGVS and using the Uniprot reference sequence P04637.">
                    Protein Description
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Location of the mutation in the introns or exons in TP53 gene for the reference sequence NM_000546.5. Terms occurring in this column are '1-intron' to '11-intron' and '2-exon' to '11-exon'. An 'i' or 'e' in front mean that the
                    mutation is located within the indicated intron or exon with no information on the precise location.">
                    Exon No
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Effect of the mutation. The terms
                                                                    occurring in this column are: missense (change of
                                                                    one
                                                                    amino-acid),
                                                                    nonsense
                                                                    (introduction of a stop codon), FS
                                                                    (frameshift), silent (no change in the protein
                                                                    sequence),
                                                                    splice (mutations located in the two first and two
                                                                    last
                                                                    conserved
                                                                    nucleotides of the introns and are thus predicted to
                                                                    alter
                                                                    splicing,
                                                                    or mutations that have been shown to alter splicing
                                                                    experimentally),
                                                                    other
                                                                    (inframe deletions or insertions), intronic
                                                                    (mutations
                                                                    in
                                                                    introns
                                                                    outside
                                                                    splicing sites), NA (mutations upstream in 5' or 3'
                                                                    UTR).">
                    Effect
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Link to NCBI SNP database.">
                    SNP Link
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Link to gnomAD database.">
                    gnomAD Link
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="Link to ClinVar database.">
                    CLINVAR Link
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="PubMed reference number provided by NCBI.">
                    PubMed
                </th>
                <th data-bs-toggle="tooltip"
                    data-bs-placement="top"
                    title="SNP databases from which the variants have been extracted.">
                    Source Databases
                </th>
            </tr>
            </thead>
            <tbody>
            {% for row in query_result.data %}
            <tr>
                <td>{{ row.g_description_GRCh38 }}</td>
                <td>{{ row.c_description }}</td>
                <td>{{ row.ProtDescription }}</td>
                <td>{{ row.ExonIntron }}</td>
                <td>{{ row.Effect }}</td>
                <td>{{ row.SNPlink }}</td>
                <td>{{ row.gnomADlink }}</td>
                <td>{{ row.CLINVARlink }}</td>
                <td>{{ row.PubMedlink }}</td>
                <td>{{ row.SourceDatabases }}</td>
            </tr>
            {% endfor %}
            </tbody>
        </table>
        {% elif table.msg %}
        <div class="row">
            <div class="col-md-12 text-center"><strong>{{ table.msg|safe }}</strong>
            </div>
        </div>
        {% else %}
        <div class="row">
            <div class="col-md-12 text-center fw-light">No data has been found
            </div>
        </div>
        {% endif %}
    </div>

</div>
</div>
{% endblock %}
